<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */
</style>


  <!--radix_placeholder_meta_tags-->
  <title>SCG: SCG analysis of Western Atlantic MAGs</title>

  <meta property="description" itemprop="description" content="Detailed workflow &amp; data products"/>

  <link rel="license" href="https://creativecommons.org/licenses/by/4.0/"/>


  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="SCG: SCG analysis of Western Atlantic MAGs"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="Detailed workflow &amp; data products"/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="SCG"/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="SCG: SCG analysis of Western Atlantic MAGs"/>
  <meta property="twitter:description" content="Detailed workflow &amp; data products"/>

  <!--/radix_placeholder_meta_tags-->
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","site"]}},"value":[{"type":"character","attributes":{},"value":["SCG analysis of Western Atlantic MAGs"]},{"type":"character","attributes":{},"value":["Detailed workflow & data products\n"]},{"type":"character","attributes":{},"value":["distill::distill_website"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>

  <script type="application/javascript">

    window.headroom_prevent_pin = false;

    window.document.addEventListener("DOMContentLoaded", function (event) {

      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');

      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });

      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });

      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>

  <style type="text/css">

  /* Theme (user-documented overrideables for nav appearance) */

  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }

  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }

  .distill-site-nav a:hover {
    color: white;
  }

  @media print {
    .distill-site-nav {
      display: none;
    }
  }

  .distill-site-header {

  }

  .distill-site-footer {

  }


  /* Site Header */

  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }

  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }


  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }

  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }

  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }

  .distill-site-header .logo {
    padding: 0;
  }

  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }

  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }



  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }


  .distill-site-header .nav-toggle {
    display: none;
  }

  .nav-dropdown {
    display: inline-block;
    position: relative;
  }

  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }

  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }

  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .nav-dropdown-active {
    display: block;
  }

  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }

  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }

  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }

  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }

  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }


  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }

  /* Site Footer */

  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }

  /* Headroom */

  d-title {
    padding-top: 6rem;
  }

  @media print {
    d-title {
      padding-top: 4rem;
    }
  }

  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }

  .headroom--transition {
    transition: all .4s ease-in-out;
  }

  .headroom--unpinned {
    top: -100px;
  }

  .headroom--pinned {
    top: 0;
  }

  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }

  </style>

  <script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure img {
    width: 100%;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refs = $(this).attr('data-cites').split(" ");
        var refHtml = refs.map(function(ref) {
          return "<p>" + $('#ref-' + ref).html() + "</p>";
        }).join("\n");
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
  <link href="site_libs/panelset-0.2.6/panelset.css" rel="stylesheet" />
  <script src="site_libs/panelset-0.2.6/panelset.js"></script>
  <script src="site_libs/clipboard-2.0.6/clipboard.min.js"></script>
  <link href="site_libs/xaringanExtra-clipboard-0.2.6/xaringanExtra-clipboard.css" rel="stylesheet" />
  <script src="site_libs/xaringanExtra-clipboard-0.2.6/xaringanExtra-clipboard.js"></script>
  <script>window.xaringanExtraClipboard(null, {"button":"<i class=\"fa fa-clone fa-2x\" style=\"color: #301e64\"><\/i>","success":"<i class=\"fa fa-check fa-2x\" style=\"color: #90BE6D\"><\/i>","error":"<i class=\"fa fa-times fa-2x\" style=\"color: #F94144\"><\/i>"})</script>
  <script src="site_libs/font-awesome-5.13.0/js/script.js"></script>
  <script src="site_libs/core-js-2.5.3/shim.min.js"></script>
  <script src="site_libs/react-17.0.0/react.min.js"></script>
  <script src="site_libs/react-17.0.0/react-dom.min.js"></script>
  <script src="site_libs/reactwidget-1.0.0/react-tools.js"></script>
  <script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
  <script src="site_libs/reactable-binding-0.2.3.9000/reactable.js"></script>
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <style type="text/css">
  /* Whole document: */
  @import url('https://fonts.googleapis.com/css?family=PT+Sans|PT+Serif|Lato:300|Inconsolata|Source+Sans+Pro|Work+Sans:200,300,400,500|Karla:400,400i,700,700i');
  /********************************/
  /* controls general properties of site*/
  /********************************/

  body {
    color: #414141;
  /*  line-height: 1.7;
    font-size: 17px; */
    background-color: #FEFEFE;
  }

  d-title h1{
  font-size: 2.1em;
  font-family: 'Work Sans';
  font-weight: 300;
  }

  d-article h1 {
    font-size: 1.8em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 300;
  }

  d-article h2 {
    font-size: 1.65em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
    font-weight: 500;
  }

  d-article h3 {
    margin-top: 1em;
    font-family: 'Work Sans', 'Source Sans Pro', 'PT Sans';
  }

  d-article a {
    color: #004276;
  }

  d-byline .authors-affiliations {
    grid-column-end: span 3;
    grid-column-gap: 8px;
  }

  img.orcid-id {
    margin: 0;
  }

  a.name:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  a.affiliation:-webkit-any-link {
    text-decoration: underline;
    text-decoration-style: dotted;
  }

  blockquote p {
  	font-size: 1.1em;
  	font-family: 'Work Sans';
    font-style: normal;
    color: #323232;
    margin: 0.5em 0 0.5em;
  }

  d-article blockquote {
    border-left: 4px solid;
    border-left-color: #323232;
  }

  d-article li {
    margin: 0px 0px 4px;
    font-size: 18px;
    font-family: 'Open Sans', 'Source Sans Pro', 'PT Sans';
  }

  /********************************/
  /* controls CODE*/
  /********************************/

  /** pre.sourceCode controls code block without lang **/

  /** Controls r code  **/

  pre.sourceCode.r {
    margin-top: 0px;
     color: #333333;
     background-color: #E8E8E8;
     border-radius: 3px;
     font-size: 0.80em;
     font-family: 'Monaco', 'PT Mono', monospace;
     overflow-x: scroll !important;
     border-left: 6px solid #AB22B2;
     border-top: 1px solid #B7B7B7;
     border-right: 1px solid #B7B7B7;
     border-bottom: 1px solid #B7B7B7;
     padding: 18px 0px 18px 20px;
   }

   pre.sourceCode.bash {
     margin-top: 0px;
      color: #333333;
      background-color: #E8E8E8;
      border-radius: 3px;
      font-size: 0.80em;
      font-family: 'Monaco', 'PT Mono', monospace;
      overflow-x: scroll !important;
      border-left: 6px solid #4A4DB5;
      border-top: 1px solid #B7B7B7;
      border-right: 1px solid #B7B7B7;
      border-bottom: 1px solid #B7B7B7;
      padding: 18px 0px 18px 20px;
    }
  /** Controls code output **/

  d-article pre {
     margin-top: 0px;
      color: #333333;
      background-color: #FAFAFA;
      font-size: 0.8em;
      border: 1px solid #ccc;
      overflow-x: scroll !important;
      padding:  10px 10px 10px 18px;
      border-radius: 5px;
    }

  /** Controls inline code  **/

  p code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }

  li code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-size: 0.8em;
    font-family: 'Monaco', 'PT Mono', monospace;
    overflow-x: scroll !important;
    margin-bottom: 20px;
    color: #BF2727;
  }

  d-article aside code {
    background-color: #E8E8E8;
    border-radius: 3px;
    font-family: 'Monaco', 'PT Mono', monospace;
    color: #BF2727;
  }

  /********************************/
  /* controls NAV BAR and HEADERS*/
  /********************************/

  .nav-dropdown-content .nav-dropdown-header {
    text-transform: none;
    color: #686868;
  }
  .nav-dropdown-content a {
    font-family: 'Work Sans'
  }
  .nav-dropdown .nav-dropbtn {
  /*  color: #FEFEFE; */
    font-weight: 400;
  }

  .distill-site-header .logo img {
    margin-top: 10px;
    max-height: 30px;
    margin-bottom: 10px;
  }

  .distill-site-nav {
      background-color: #323232;
      font-size: 16px;
      font-weight: 300;
  }

  img {
    margin: 5px;
  }

  /** LIGHTBOX MARKUP **/
  .thumbnail {
    max-width: 100%;
    position: relative;
  }

  /** DETAILS fold down**/

  details {
    background-color: #f6f6f6;
    cursor: pointer;
    color: #2d2d2d;
    margin-bottom:1vh;
    padding: 1em;
    border: none;
    text-align: left;
    outline: none;
  }

  .active, details:hover {
    transition: max-height 0.2s ease-out;
    outline: none;

    /* background-color: #acc2c2; */
  }

  details {
    display: flex;
    justify-content: space-between;
    align-content: center;
    border-radius: 4px;
    outline: none;
  }

  details.active {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    outline: none;
  }

  summary::-webkit-details-marker {
    display: none;
    outline: none;
  }

  summary:focus {
    outline: none;
  }

  summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f0fe";
  }

  details[open] summary::after {
    float: right;
    font-family: "Font Awesome 5 Free";
    content: "\f146";
  }

  /********************************/
  /* controls downloadthis  */
  /********************************/

  button.btn.btn-default {
    font-size: 0.8em;
    font-family: "PT Sans";
    border-color: #808080;
    border-width: medium;
    margin-bottom: 15px;
  }

  button.btn.btn-success {
    font-size: 1em;
    font-family: "PT Sans";
    border-color: #d55e00;
    border-width: medium;
    margin-bottom: 15px;
  }
  /********************************/
  /* controls reactable  */
  /********************************/

  div.rt-text-content {
    font-size: 1.2em;
  }

  /********************************/
  /* controls horizontal rule, ie ***  */
  /********************************/

  d-article hr {
    margin: 30px 0px;
  }

  /* .blackbox:before {
      font-family: "Font Awesome 5 Free";
      font-weight: 900;
      content: "\f15b";
      display: inline-block;
      padding-right: 3px;
      vertical-align: middle;
   }
  */
  /********************************/
  /* controls data callout box  */
  /********************************/

  .infobox {
    padding: 1em;
    background: #e8dbde;
    color: #323232;
    border: 1px solid black;
    border-radius: 10px;
    margin-top: 1em;
    margin-bottom: 1em;
    position: relative;
  }

  .infobox p {
    margin: 0px 0px 0px;
    font-size: 1em;
  }

  .center {
    text-align: center;
    font-size: 0.9em;
  }

  </style>
  <!--/radix_placeholder_site_in_header-->

  <link rel="stylesheet" href="assets/css/styles.css" type="text/css"/>

</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"SCG analysis of Western Atlantic MAGs","description":"Detailed workflow & data products","authors":[]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title"></a>
</div>
<div class="nav-right">
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>SCG analysis of Western Atlantic MAGs</h1>
<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->
<p><p>Detailed workflow &amp; data products</p></p>
</div>


<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#mag-workflow">MAG Workflow</a>
<ul>
<li><a href="#description-of-data-products">Description of Data Products</a></li>
<li><a href="#generate-scg-lists">Generate SCG lists</a></li>
<li><a href="#mag-completeness-taxonomy">MAG Completeness &amp; Taxonomy</a></li>
<li><a href="#scg-hits-across-mags">SCG Hits across MAGs</a></li>
<li><a href="#scg-coverage-across-mags">SCG Coverage across MAGs</a></li>
</ul></li>
<li><a href="#metagenome-workflow">Metagenome Workflow</a>
<ul>
<li><a href="#description-of-data-products-1">Description of Data Products</a></li>
<li><a href="#generate-scg-lists-1">Generate SCG lists</a></li>
<li><a href="#scg-coverage">SCG Coverage</a></li>
<li><a href="#gtdb-scg-taxonomy">GTDB SCG Taxonomy</a></li>
</ul></li>
</ul>
</nav>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="mag-workflow">MAG Workflow</h1>
<p>In the first section of this workflow, we assess the taxonomy and coverage of SCG from our MAG collection. In the second part, we assess SCG across the entire metagenome.</p>
<h2 id="description-of-data-products">Description of Data Products</h2>
<p>The link below gives access to all data and data products generated in the workflow. This is a 7.5Mb zip file.</p>
<div class="layout-chunk" data-layout="l-body">
<a href="https://github.com/istmobiome/scg/raw/main/files/MAG_WF_DATA.zip">
<button class="btn btn-success"><i class="fa fa-save"></i> Download the full data pack here</button>
</a>
</div>
<blockquote>
<p>Note: the sub directiories <strong>12_COVERAGE</strong> and <strong>12_DETECTION</strong> contain the final coverage/detection data for each MAG. See below for a description of all data products.</p>
</blockquote>
<p>There are a number of data products generated in this workflow. Below is a description of each piece from this puzzle and links to download the data. Some of this information is redundant, others may not be useful, but all are included here for posterity.</p>
<p>First, a description of individual tables.</p>
<div class="l-body-outset">
<table>
<colgroup>
<col style="width: 59%" />
<col style="width: 40%" />
</colgroup>
<thead>
<tr class="header">
<th><strong>file name</strong></th>
<th><strong>description </strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><a href="files/MAG/00_WA-MAGS.txt">00_WA-MAGS.txt</a></td>
<td>list of all MAGs in the collection.</td>
</tr>
<tr class="even">
<td><a href="files/MAG/01_hmm_hits_info.txt">01_hmm_hits_info.txt</a></td>
<td>HMM profiles in the contigs database</td>
</tr>
<tr class="odd">
<td><a href="files/MAG/02_Archaea_76.txt">02_Archaea_76.txt</a></td>
<td>description of Archaea_76 HMM profile.</td>
</tr>
<tr class="even">
<td><a href="files/MAG/02_Bacteria_71.txt">02_Bacteria_71.txt</a></td>
<td>description of Bacteria_71 HMM profile.</td>
</tr>
<tr class="odd">
<td><a href="files/MAG/03_GTDB_Taxonomy.txt">03_GTDB_Taxonomy.txt</a></td>
<td>list of GTDB taxonomy genes.</td>
</tr>
<tr class="even">
<td><a href="files/MAG/04_SCG_master_list.txt">04_SCG_master_list.txt</a></td>
<td>master list of all SCG used in this analysis.</td>
</tr>
<tr class="odd">
<td><a href="files/MAG/05_MAG-COMPLETENESS.txt">05_MAG-COMPLETENESS.txt</a></td>
<td>completion &amp; genome statistics for each MAG.</td>
</tr>
<tr class="even">
<td><a href="files/MAG/05_MAG-TAXONOMY.txt">05_MAG-TAXONOMY.txt</a></td>
<td>GTDB taxonomy for each MAG.</td>
</tr>
<tr class="odd">
<td><a href="files/MAG/05_MAG-TAXONOMY-PER-GENOME-SCG.txt">05_MAG-TAXONOMY-PER-GENOME-SCG.txt</a></td>
<td>GTDB taxonomy for each gene in a MAG.</td>
</tr>
<tr class="even">
<td><a href="files/MAG/06_MAG_TAXONOMY_and_COMPLETENESS.txt">06_MAG_TAXONOMY_and_COMPLETENESS.txt</a></td>
<td>combined completion &amp; taxonomy estimates.</td>
</tr>
<tr class="odd">
<td><a href="files/MAG/07_Archaea_76-across-MAGs.txt">07_Archaea_76-across-MAGs.txt</a></td>
<td>distribution of Archaea_76 SCG across MAGs.</td>
</tr>
<tr class="even">
<td><a href="files/MAG/07_Bacteria_71-across-MAGs.txt">07_Bacteria_71-across-MAGs.txt</a></td>
<td>distribution of Bacteria_71 SCG across MAGs.</td>
</tr>
<tr class="odd">
<td><a href="files/MAG/08_MAGS-Archaea_76-sequences.fa">08_MAGS-Archaea_76-sequences.fa</a></td>
<td>fasta file of Archaea_76 SCG hits across MAGs.</td>
</tr>
<tr class="even">
<td><a href="files/MAG/08_MAGS-Bacteria_71-sequences.fa">08_MAGS-Bacteria_71-sequences.fa</a></td>
<td>fasta file of Bacteria_71 SCG hits across MAGs.</td>
</tr>
<tr class="odd">
<td><a href="files/MAG/09_MAGS-Archaea_76-details.txt">09_MAGS-Archaea_76-details.txt</a></td>
<td>parsed SCG details from Archaea_76 fasta file.</td>
</tr>
<tr class="even">
<td><a href="files/MAG/09_MAGS-Bacteria_71-details.txt">09_MAGS-Bacteria_71-details.txt</a></td>
<td>parsed SCG details from Bacteria_71 fasta file.</td>
</tr>
<tr class="odd">
<td><a href="files/MAG/internal-genomes.txt">internal-genomes.txt</a></td>
<td>paths to CONTIGS.db &amp; PROFILE.db for each MAG.</td>
</tr>
</tbody>
</table>
</div>
<div class="l-body-outset">
<p>Now a description of the data from the main results of the analysis. Each main directory is named by the HMM profile (<code>Archaea_76/</code> or <code>Bacteria_71/</code>) and both contain the following sub directories:</p>
<table>
<colgroup>
<col style="width: 27%" />
<col style="width: 72%" />
</colgroup>
<thead>
<tr class="header">
<th><strong>subdirectory</strong></th>
<th><strong>description </strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>09_HMM_HITS_PER_GENE_CALLS</td>
<td>details of SCG present in each MAG</td>
</tr>
<tr class="even">
<td>10_GENE_CALLER_IDS</td>
<td>list of SCG IDs for each MAG</td>
</tr>
<tr class="odd">
<td>11_COVERAGE_RAW</td>
<td>raw output of coverage data from <code>anvi-script-get-hmm-hits-per-gene-call</code></td>
</tr>
<tr class="even">
<td>11_DETECTION_RAW</td>
<td>raw output of detection data from <code>anvi-script-get-hmm-hits-per-gene-call</code></td>
</tr>
<tr class="odd">
<td><strong>12_COVERAGE</strong></td>
<td><strong>final output of coverage data for each MAG</strong></td>
</tr>
<tr class="even">
<td><strong>12_DETECTION</strong></td>
<td><strong>final output of detection data for each MAG</strong></td>
</tr>
</tbody>
</table>
</div>
<h2 id="generate-scg-lists">Generate SCG lists</h2>
<p>The first step is to determine which HMM profiles are present in the <code>contigs.db</code>. There are a few different ways to accomplish this task. Here we run the following command:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-export-table</span> 03_CONTIGS/WA-contigs.db <span class="at">--table</span> hmm_hits_info <span class="dt">\</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--output-file</span> 01_hmm_hits_info.txt</span></code></pre></div>
<p>This command produces a file that contains information about each HMM profile in the <code>contigs.db</code>. Here I modified the output table a bit to include links to the databases.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><small><strong>Table 1 | Detailed description of all HMM profiles contained in the WA contigs database, including gene IDs and database links. </strong></small></p>
<div class="layout-chunk" data-layout="l-page">
<div id="htmlwidget-b3db68725ed0b51a2fc0" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-b3db68725ed0b51a2fc0">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"source":["Archaea_76","Bacteria_71","Protista_83","Ribosomal_RNA_12S","Ribosomal_RNA_16S","Ribosomal_RNA_18S","Ribosomal_RNA_23S","Ribosomal_RNA_28S","Ribosomal_RNA_5S","Transfer_RNAs"],"reference":["Lee","Lee modified","Delmont","Seeman T","Seeman T","Seeman T","Seeman T","Seeman T","Seeman T","Chan and Lowe"],"doi":["https://doi.org/10.1093/bioinformatics/btz188","https://doi.org/10.1093/bioinformatics/btz188","http://merenlab.org/delmont-euk-scgs","https://github.com/tseemann/barrnap","https://github.com/tseemann/barrnap","https://github.com/tseemann/barrnap","https://github.com/tseemann/barrnap","https://github.com/tseemann/barrnap","https://github.com/tseemann/barrnap","https://doi.org/10.1007/978-1-4939-9173-0_1"],"db_link":["https://github.com/meren/anvio/tree/master/anvio/data/hmm/Archaea_76","https://github.com/meren/anvio/tree/master/anvio/data/hmm/Bacteria_71","https://github.com/meren/anvio/tree/master/anvio/data/hmm/Protista_83","https://github.com/meren/anvio/tree/master/anvio/data/hmm/Ribosomal_RNA_12S","https://github.com/meren/anvio/tree/master/anvio/data/hmm/Ribosomal_RNA_16S","https://github.com/meren/anvio/tree/master/anvio/data/hmm/Ribosomal_RNA_18S","https://github.com/meren/anvio/tree/master/anvio/data/hmm/Ribosomal_RNA_23S","https://github.com/meren/anvio/tree/master/anvio/data/hmm/Ribosomal_RNA_28S","https://github.com/meren/anvio/tree/master/anvio/data/hmm/Ribosomal_RNA_5S",null],"search_type":["singlecopy","singlecopy","singlecopy","Ribosomal_RNA_12S","Ribosomal_RNA_16S","Ribosomal_RNA_18S","Ribosomal_RNA_23S","Ribosomal_RNA_28S","Ribosomal_RNA_5S","Transfer_RNAs"],"domain":["archaea","bacteria","eukarya",null,null,null,null,null,null,null],"genes":["Adenylsucc_synt, AdoHcyase, Archease, ATP-synt_D, ATP-synt_F, CarS-like, CTP-dep_RFKase, Diphthamide_syn, DNA_primase_lrg, dsDNA_bind, DUF357, DUF359, DUF655, eIF-6, FbpA, Ham1p_like, HMG-CoA_red, NDK, PPS_PS, Prefoldin, PTH2, PyrI, Ribosomal_L1, Ribosomal_L13, Ribosomal_L14, Ribosomal_L15e, Ribosomal_L16, Ribosomal_L21e, Ribosomal_L22, Ribosomal_L23, Ribosomal_L26, Ribosomal_L29, Ribosomal_L3, Ribosomal_L31e, Ribosomal_L32e, Ribosomal_L37ae, Ribosomal_L39, Ribosomal_L4, Ribosomal_L44, Ribosomal_L5e, Ribosomal_L6, Ribosomal_S11, Ribosomal_S13, Ribosomal_S15, Ribosomal_S17, Ribosomal_S17e, Ribosomal_S19, Ribosomal_S19e, Ribosomal_S2, Ribosomal_S24e, Ribosomal_S27e, Ribosomal_S28e, Ribosomal_S3Ae, Ribosomal_S7, Ribosomal_S8, Ribosomal_S8e, Ribosomal_S9, Ribosom_S12_S23, Rib_5-P_isom_A, RNase_HII, RNA_pol_L_2, RNA_pol_N, RNA_pol_Rpb4, RNA_pol_Rpb6, RtcB, SecY, Spt4, TIM, Trm56, tRNA-synt_1c, tRNA-synt_1d, tRNA-synt_His, TruD, vATP-synt_AC39, vATP-synt_E, V_ATPase_I","Adenylsucc_synt, ADK, AICARFT_IMPCHas, ATP-synt, ATP-synt_A, Chorismate_synt, EF_TS, eIF-1a, Exonuc_VII_L, GrpE, Ham1p_like, IPPT, OSCP, Pept_tRNA_hydro, PGK, RBFA, RecO_C, Ribonuclease_P, Ribosomal_L1, Ribosomal_L2, Ribosomal_L13, Ribosomal_L14, Ribosomal_L16, Ribosomal_L17, Ribosomal_L18p, Ribosomal_L19, Ribosomal_L20, Ribosomal_L21p, Ribosomal_L22, Ribosomal_L23, ribosomal_L24, Ribosomal_L27, Ribosomal_L27A, Ribosomal_L28, Ribosomal_L29, Ribosomal_L3, Ribosomal_L32p, Ribosomal_L35p, Ribosomal_L4, Ribosomal_L5, Ribosomal_L6, Ribosomal_L9_C, Ribosomal_S10, Ribosomal_S11, Ribosomal_S13, Ribosomal_S15, Ribosomal_S16, Ribosomal_S17, Ribosomal_S19, Ribosomal_S2, Ribosomal_S3_C, Ribosomal_S6, Ribosomal_S7, Ribosomal_S8, Ribosomal_S9, Ribosomal_S20p, Ribosom_S12_S23, RNA_pol_L, RNA_pol_Rpb6, RRF, RsfS, RuvX, SecE, SecG, SecY, SmpB, tRNA-synt_1d, tRNA_m1G_MT, TsaE, UPF0054, YajC","EPrGT00050000000570, EPrGT00050000000597, EPrGT00050000000633, EPrGT00050000000638, EPrGT00050000000656, EPrGT00050000000764, EPrGT00050000000788, EPrGT00050000000790, EPrGT00050000000821, EPrGT00050000000974, EPrGT00050000001261, EPrGT00050000001346, EPrGT00050000001368, EPrGT00050000001885, EPrGT00050000001927, EPrGT00050000002128, EPrGT00050000002199, EPrGT00050000002214, EPrGT00050000002346, EPrGT00050000002389, EPrGT00050000002690, EPrGT00050000003076, EPrGT00050000003160, EPrGT00050000003199, EPrGT00050000003303, EPrGT00050000003367, EPrGT00050000003603, EPrGT00050000003615, EPrGT00050000003868, EPrGT00050000004000, EPrGT00050000004283, EPrGT00050000004299, EPrGT00050000004774, EPrGT00050000004932, EPrGT00050000005003, EPrGT00050000005020, EPrGT00050000005111, EPrGT00050000005128, EPrGT00050000005182, EPrGT00050000005194, EPrGT00050000005290, EPrGT00050000005370, EPrGT00050000005427, EPrGT00050000005459, EPrGT00050000005482, EPrGT00050000005485, EPrGT00050000005498, EPrGT00050000005502, EPrGT00050000005514, EPrGT00050000005516, EPrGT00050000005533, EPrGT00050000005562, EPrGT00050000005659, EPrGT00050000005687, EPrGT00050000005714, EPrGT00050000005721, EPrGT00050000005732, EPrGT00050000005743, EPrGT00050000005769, EPrGT00050000005798, EPrGT00050000005800, EPrGT00050000005808, EPrGT00050000005840, EPrGT00050000005852, EPrGT00050000005857, EPrGT00050000005939, EPrGT00050000005940, EPrGT00050000005954, EPrGT00050000005966, EPrGT00050000005995, EPrGT00050000006007, EPrGT00050000006026, EPrGT00050000006045, EPrGT00050000006077, EPrGT00050000006107, EPrGT00050000006117, EPrGT00050000006133, EPrGT00050000007812, EPrGT00080000010433, EPrGT00180000031570, EPrGT00300000062292, EPrGT00300000062307, EPrGT00300000062326","12S_rRNA_mit","16S_rRNA_bac, 16S_rRNA_arc, 16S_rRNA_mit","18S_rRNA_euk","23S_rRNA_bac, 23S_rRNA_arc","28S_rRNA_euk","5S_rRNA_arc, 5S_rRNA_euk, 5S_rRNA_bac, 5_8S_rRNA_euk, 5_8S_rRNA_arc","Pro_GGG, Cys_GCA, Arg_ACG, Cys_ACA, Ser_AGA, Arg_TCG, Leu_CAG, Tyr_ATA, Glu_CTC, Pro_CGG, His_GTG, Tyr_GTA, Pro_TGG, His_ATG, Phe_AAA, Gly_CCC, Leu_TAG, Lys_TTT, Gly_ACC, Met_CAT, Ile_AAT, Ile_TAT, Phe_GAA, Asp_GTC, Thr_AGT, Lys_CTT, Ser_ACT, Ala_AGC, Ser_CGA, Val_GAC, Ser_TGA, Gly_TCC, Gln_TTG, Arg_GCG, Ser_GGA, Leu_AAG, Ala_GGC, Arg_TCT, Gln_CTG, Ala_TGC, Asn_ATT, Ser_GCT, Asn_GTT, Ala_CGC, Ile_GAT, Thr_CGT, Trp_CCA, Arg_CCT, Thr_TGT, Arg_CCG, Thr_GGT, Glu_TTC, Asp_ATC, Val_CAC, Leu_TAA, Leu_GAG, Leu_CAA, Val_TAC, Pro_AGG, Gly_GCC, Val_AAC"]},"columns":[{"accessor":"source","name":"Source","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["Archaea_76","Bacteria_71","Protista_83","Ribosomal_RNA_12S","Ribosomal_RNA_16S","Ribosomal_RNA_18S","Ribosomal_RNA_23S","Ribosomal_RNA_28S","Ribosomal_RNA_5S","Transfer_RNAs"],"header":"Source","align":"left","footerStyle":{"fontWeight":"bold"},"minWidth":200,"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"reference","name":"reference","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["<a href=\"https://doi.org/10.1093/bioinformatics/btz188\" target=\"_blank\">Lee<\/a>","<a href=\"https://doi.org/10.1093/bioinformatics/btz188\" target=\"_blank\">Lee modified<\/a>","<a href=\"http://merenlab.org/delmont-euk-scgs\" target=\"_blank\">Delmont<\/a>","<a href=\"https://github.com/tseemann/barrnap\" target=\"_blank\">Seeman T<\/a>","<a href=\"https://github.com/tseemann/barrnap\" target=\"_blank\">Seeman T<\/a>","<a href=\"https://github.com/tseemann/barrnap\" target=\"_blank\">Seeman T<\/a>","<a href=\"https://github.com/tseemann/barrnap\" target=\"_blank\">Seeman T<\/a>","<a href=\"https://github.com/tseemann/barrnap\" target=\"_blank\">Seeman T<\/a>","<a href=\"https://github.com/tseemann/barrnap\" target=\"_blank\">Seeman T<\/a>","<a href=\"https://doi.org/10.1007/978-1-4939-9173-0_1\" target=\"_blank\">Chan and Lowe<\/a>"],"header":"reference","align":"left","footerStyle":{"fontWeight":"bold"},"html":true,"minWidth":150},{"accessor":"doi","name":"doi","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":[{"name":"a","attribs":{"href":"https://doi.org/10.1093/bioinformatics/btz188","target":"_blank"},"children":["https://doi.org/10.1093/bioinformatics/btz188"]},{"name":"a","attribs":{"href":"https://doi.org/10.1093/bioinformatics/btz188","target":"_blank"},"children":["https://doi.org/10.1093/bioinformatics/btz188"]},{"name":"a","attribs":{"href":"http://merenlab.org/delmont-euk-scgs","target":"_blank"},"children":["http://merenlab.org/delmont-euk-scgs"]},{"name":"a","attribs":{"href":"https://github.com/tseemann/barrnap","target":"_blank"},"children":["https://github.com/tseemann/barrnap"]},{"name":"a","attribs":{"href":"https://github.com/tseemann/barrnap","target":"_blank"},"children":["https://github.com/tseemann/barrnap"]},{"name":"a","attribs":{"href":"https://github.com/tseemann/barrnap","target":"_blank"},"children":["https://github.com/tseemann/barrnap"]},{"name":"a","attribs":{"href":"https://github.com/tseemann/barrnap","target":"_blank"},"children":["https://github.com/tseemann/barrnap"]},{"name":"a","attribs":{"href":"https://github.com/tseemann/barrnap","target":"_blank"},"children":["https://github.com/tseemann/barrnap"]},{"name":"a","attribs":{"href":"https://github.com/tseemann/barrnap","target":"_blank"},"children":["https://github.com/tseemann/barrnap"]},{"name":"a","attribs":{"href":"https://doi.org/10.1007/978-1-4939-9173-0_1","target":"_blank"},"children":["https://doi.org/10.1007/978-1-4939-9173-0_1"]}],"header":"doi","align":"left","footerStyle":{"fontWeight":"bold"},"show":false},{"accessor":"db_link","name":"db_link","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":[{"name":"a","attribs":{"href":"https://github.com/meren/anvio/tree/master/anvio/data/hmm/Archaea_76","target":"_blank"},"children":["https://github.com/meren/anvio/tree/master/anvio/data/hmm/Archaea_76"]},{"name":"a","attribs":{"href":"https://github.com/meren/anvio/tree/master/anvio/data/hmm/Bacteria_71","target":"_blank"},"children":["https://github.com/meren/anvio/tree/master/anvio/data/hmm/Bacteria_71"]},{"name":"a","attribs":{"href":"https://github.com/meren/anvio/tree/master/anvio/data/hmm/Protista_83","target":"_blank"},"children":["https://github.com/meren/anvio/tree/master/anvio/data/hmm/Protista_83"]},{"name":"a","attribs":{"href":"https://github.com/meren/anvio/tree/master/anvio/data/hmm/Ribosomal_RNA_12S","target":"_blank"},"children":["https://github.com/meren/anvio/tree/master/anvio/data/hmm/Ribosomal_RNA_12S"]},{"name":"a","attribs":{"href":"https://github.com/meren/anvio/tree/master/anvio/data/hmm/Ribosomal_RNA_16S","target":"_blank"},"children":["https://github.com/meren/anvio/tree/master/anvio/data/hmm/Ribosomal_RNA_16S"]},{"name":"a","attribs":{"href":"https://github.com/meren/anvio/tree/master/anvio/data/hmm/Ribosomal_RNA_18S","target":"_blank"},"children":["https://github.com/meren/anvio/tree/master/anvio/data/hmm/Ribosomal_RNA_18S"]},{"name":"a","attribs":{"href":"https://github.com/meren/anvio/tree/master/anvio/data/hmm/Ribosomal_RNA_23S","target":"_blank"},"children":["https://github.com/meren/anvio/tree/master/anvio/data/hmm/Ribosomal_RNA_23S"]},{"name":"a","attribs":{"href":"https://github.com/meren/anvio/tree/master/anvio/data/hmm/Ribosomal_RNA_28S","target":"_blank"},"children":["https://github.com/meren/anvio/tree/master/anvio/data/hmm/Ribosomal_RNA_28S"]},{"name":"a","attribs":{"href":"https://github.com/meren/anvio/tree/master/anvio/data/hmm/Ribosomal_RNA_5S","target":"_blank"},"children":["https://github.com/meren/anvio/tree/master/anvio/data/hmm/Ribosomal_RNA_5S"]},{"name":"a","attribs":{"href":null,"target":"_blank"},"children":["NA"]}],"header":"db link","align":"left","footerStyle":{"fontWeight":"bold"},"minWidth":200},{"accessor":"search_type","name":"search_type","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["singlecopy","singlecopy","singlecopy","Ribosomal_RNA_12S","Ribosomal_RNA_16S","Ribosomal_RNA_18S","Ribosomal_RNA_23S","Ribosomal_RNA_28S","Ribosomal_RNA_5S","Transfer_RNAs"],"header":"search type","align":"left","footerStyle":{"fontWeight":"bold"},"minWidth":200,"sticky":"left"},{"accessor":"domain","name":"domain","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["archaea","bacteria","eukarya","NA","NA","NA","NA","NA","NA","NA"],"header":"domain","align":"left","footerStyle":{"fontWeight":"bold"},"minWidth":90,"sticky":"left"},{"accessor":"genes","name":"genes","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["Adenylsucc_synt, AdoHcyase, Archease, ATP-synt_D, ATP-synt_F, CarS-like, CTP-dep_RFKase, Diphthamide_syn, DNA_primase_lrg, dsDNA_bind, DUF357, DUF359, DUF655, eIF-6, FbpA, Ham1p_like, HMG-CoA_red, NDK, PPS_PS, Prefoldin, PTH2, PyrI, Ribosomal_L1, Ribosomal_L13, Ribosomal_L14, Ribosomal_L15e, Ribosomal_L16, Ribosomal_L21e, Ribosomal_L22, Ribosomal_L23, Ribosomal_L26, Ribosomal_L29, Ribosomal_L3, Ribosomal_L31e, Ribosomal_L32e, Ribosomal_L37ae, Ribosomal_L39, Ribosomal_L4, Ribosomal_L44, Ribosomal_L5e, Ribosomal_L6, Ribosomal_S11, Ribosomal_S13, Ribosomal_S15, Ribosomal_S17, Ribosomal_S17e, Ribosomal_S19, Ribosomal_S19e, Ribosomal_S2, Ribosomal_S24e, Ribosomal_S27e, Ribosomal_S28e, Ribosomal_S3Ae, Ribosomal_S7, Ribosomal_S8, Ribosomal_S8e, Ribosomal_S9, Ribosom_S12_S23, Rib_5-P_isom_A, RNase_HII, RNA_pol_L_2, RNA_pol_N, RNA_pol_Rpb4, RNA_pol_Rpb6, RtcB, SecY, Spt4, TIM, Trm56, tRNA-synt_1c, tRNA-synt_1d, tRNA-synt_His, TruD, vATP-synt_AC39, vATP-synt_E, V_ATPase_I","Adenylsucc_synt, ADK, AICARFT_IMPCHas, ATP-synt, ATP-synt_A, Chorismate_synt, EF_TS, eIF-1a, Exonuc_VII_L, GrpE, Ham1p_like, IPPT, OSCP, Pept_tRNA_hydro, PGK, RBFA, RecO_C, Ribonuclease_P, Ribosomal_L1, Ribosomal_L2, Ribosomal_L13, Ribosomal_L14, Ribosomal_L16, Ribosomal_L17, Ribosomal_L18p, Ribosomal_L19, Ribosomal_L20, Ribosomal_L21p, Ribosomal_L22, Ribosomal_L23, ribosomal_L24, Ribosomal_L27, Ribosomal_L27A, Ribosomal_L28, Ribosomal_L29, Ribosomal_L3, Ribosomal_L32p, Ribosomal_L35p, Ribosomal_L4, Ribosomal_L5, Ribosomal_L6, Ribosomal_L9_C, Ribosomal_S10, Ribosomal_S11, Ribosomal_S13, Ribosomal_S15, Ribosomal_S16, Ribosomal_S17, Ribosomal_S19, Ribosomal_S2, Ribosomal_S3_C, Ribosomal_S6, Ribosomal_S7, Ribosomal_S8, Ribosomal_S9, Ribosomal_S20p, Ribosom_S12_S23, RNA_pol_L, RNA_pol_Rpb6, RRF, RsfS, RuvX, SecE, SecG, SecY, SmpB, tRNA-synt_1d, tRNA_m1G_MT, TsaE, UPF0054, YajC","EPrGT00050000000570, EPrGT00050000000597, EPrGT00050000000633, EPrGT00050000000638, EPrGT00050000000656, EPrGT00050000000764, EPrGT00050000000788, EPrGT00050000000790, EPrGT00050000000821, EPrGT00050000000974, EPrGT00050000001261, EPrGT00050000001346, EPrGT00050000001368, EPrGT00050000001885, EPrGT00050000001927, EPrGT00050000002128, EPrGT00050000002199, EPrGT00050000002214, EPrGT00050000002346, EPrGT00050000002389, EPrGT00050000002690, EPrGT00050000003076, EPrGT00050000003160, EPrGT00050000003199, EPrGT00050000003303, EPrGT00050000003367, EPrGT00050000003603, EPrGT00050000003615, EPrGT00050000003868, EPrGT00050000004000, EPrGT00050000004283, EPrGT00050000004299, EPrGT00050000004774, EPrGT00050000004932, EPrGT00050000005003, EPrGT00050000005020, EPrGT00050000005111, EPrGT00050000005128, EPrGT00050000005182, EPrGT00050000005194, EPrGT00050000005290, EPrGT00050000005370, EPrGT00050000005427, EPrGT00050000005459, EPrGT00050000005482, EPrGT00050000005485, EPrGT00050000005498, EPrGT00050000005502, EPrGT00050000005514, EPrGT00050000005516, EPrGT00050000005533, EPrGT00050000005562, EPrGT00050000005659, EPrGT00050000005687, EPrGT00050000005714, EPrGT00050000005721, EPrGT00050000005732, EPrGT00050000005743, EPrGT00050000005769, EPrGT00050000005798, EPrGT00050000005800, EPrGT00050000005808, EPrGT00050000005840, EPrGT00050000005852, EPrGT00050000005857, EPrGT00050000005939, EPrGT00050000005940, EPrGT00050000005954, EPrGT00050000005966, EPrGT00050000005995, EPrGT00050000006007, EPrGT00050000006026, EPrGT00050000006045, EPrGT00050000006077, EPrGT00050000006107, EPrGT00050000006117, EPrGT00050000006133, EPrGT00050000007812, EPrGT00080000010433, EPrGT00180000031570, EPrGT00300000062292, EPrGT00300000062307, EPrGT00300000062326","12S_rRNA_mit","16S_rRNA_bac, 16S_rRNA_arc, 16S_rRNA_mit","18S_rRNA_euk","23S_rRNA_bac, 23S_rRNA_arc","28S_rRNA_euk","5S_rRNA_arc, 5S_rRNA_euk, 5S_rRNA_bac, 5_8S_rRNA_euk, 5_8S_rRNA_arc","Pro_GGG, Cys_GCA, Arg_ACG, Cys_ACA, Ser_AGA, Arg_TCG, Leu_CAG, Tyr_ATA, Glu_CTC, Pro_CGG, His_GTG, Tyr_GTA, Pro_TGG, His_ATG, Phe_AAA, Gly_CCC, Leu_TAG, Lys_TTT, Gly_ACC, Met_CAT, Ile_AAT, Ile_TAT, Phe_GAA, Asp_GTC, Thr_AGT, Lys_CTT, Ser_ACT, Ala_AGC, Ser_CGA, Val_GAC, Ser_TGA, Gly_TCC, Gln_TTG, Arg_GCG, Ser_GGA, Leu_AAG, Ala_GGC, Arg_TCT, Gln_CTG, Ala_TGC, Asn_ATT, Ser_GCT, Asn_GTT, Ala_CGC, Ile_GAT, Thr_CGT, Trp_CCA, Arg_CCT, Thr_TGT, Arg_CCG, Thr_GGT, Glu_TTC, Asp_ATC, Val_CAC, Leu_TAA, Leu_GAG, Leu_CAA, Val_TAC, Pro_AGG, Gly_GCC, Val_AAC"],"header":"genes","align":"left","footerStyle":{"fontWeight":"bold"},"minWidth":2000,"sticky":"left"}],"defaultPageSize":10,"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"b0a8411f568a1ad3699236acfe43d15d"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
<div class="infobox">
<div class="center">
<p><i class="fas fa-file"></i> <strong>Data pack file</strong></p>
</div>
<p>01_hmm_hits_info.txt</p>
</div>
<p>For this analysis we are mainly interested in the <code>Archaea_76</code> and <code>Bacteria_71</code> profiles. Here we need to parse the <code>genes</code> column for both sources and create a master list of target single copy genes (SCG) for each.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Archaea_76</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>tmp_tbl <span class="ot">&lt;-</span> hmm_hits_info[<span class="dv">1</span>, <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">7</span>)]</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>tmp_tbl <span class="ot">&lt;-</span> tidyr<span class="sc">::</span><span class="fu">separate_rows</span>(tmp_tbl, <span class="st">&quot;genes&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;, &quot;</span>, <span class="at">convert =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>tmp_tbl[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span> </span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>arch_tbl <span class="ot">&lt;-</span> tmp_tbl </span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Bacteria_71</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>tmp_tbl <span class="ot">&lt;-</span> hmm_hits_info[<span class="dv">2</span>, <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">7</span>)]</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>tmp_tbl <span class="ot">&lt;-</span> tidyr<span class="sc">::</span><span class="fu">separate_rows</span>(tmp_tbl, <span class="st">&quot;genes&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;, &quot;</span>, <span class="at">convert =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>tmp_tbl[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span> </span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>bact_tbl <span class="ot">&lt;-</span> tmp_tbl </span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(tmp_tbl)</span></code></pre></div>
</details>
</div>
<p>It would also be nice to include the <a href="https://pfam.xfam.org/">Pfam</a> ID and database links for each SCG. We can get this information from the anvi’o database links provided in the table above.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>tmp_file <span class="ot">&lt;-</span> <span class="st">&#39;https://raw.githubusercontent.com/meren/anvio/master/anvio/data/hmm/Archaea_76/genes.txt&#39;</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(tmp_file, <span class="at">destfile =</span> <span class="st">&quot;files/MAG/02_Archaea_76.txt&quot;</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(tmp_file)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>tmp_file <span class="ot">&lt;-</span> <span class="st">&#39;https://raw.githubusercontent.com/meren/anvio/master/anvio/data/hmm/Bacteria_71/genes.txt&#39;</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(tmp_file, <span class="at">destfile =</span> <span class="st">&quot;files/MAG/02_Bacteria_71.txt&quot;</span>)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(tmp_file)</span></code></pre></div>
</details>
</div>
<p>Then we add the prefix <code>https://pfam.xfam.org/family/</code> to each accession number to generate a hyperlink to the Pfam database.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Archaea_76</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>tmp_tbl <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">&quot;files/MAG/02_Archaea_76.txt&quot;</span>, <span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>tmp_tbl<span class="sc">$</span>hmmsource <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>tmp_tbl<span class="sc">$</span>pfam_link <span class="ot">&lt;-</span> tmp_tbl<span class="sc">$</span>accession</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>tmp_tbl<span class="sc">$</span>pfam_link <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">&#39;https://pfam.xfam.org/family/&#39;</span>, tmp_tbl<span class="sc">$</span>pfam_link)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>arch_tbl <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(arch_tbl, tmp_tbl, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;genes&quot;</span> <span class="ot">=</span> <span class="st">&quot;gene&quot;</span>))</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(tmp_tbl)</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Bacteria_71</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>tmp_tbl <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">&quot;files/MAG/02_Bacteria_71.txt&quot;</span>, <span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>tmp_tbl<span class="sc">$</span>hmmsource <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>tmp_tbl<span class="sc">$</span>pfam_link <span class="ot">&lt;-</span> tmp_tbl<span class="sc">$</span>accession</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>tmp_tbl<span class="sc">$</span>pfam_link <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">&#39;https://pfam.xfam.org/family/&#39;</span>, tmp_tbl<span class="sc">$</span>pfam_link)</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>bact_tbl <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(bact_tbl, tmp_tbl, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;genes&quot;</span> <span class="ot">=</span> <span class="st">&quot;gene&quot;</span>))</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(tmp_tbl)</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="st">&quot;rm files/MAG/02_Archaea_76.txt&quot;</span>)</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="st">&quot;rm files/MAG/02_Bacteria_71.txt&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>And then save the tables for each profile.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(arch_tbl, <span class="st">&quot;files/MAG/02_Archaea_76.txt&quot;</span>, <span class="at">col.names  =</span> <span class="cn">TRUE</span>, </span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(bact_tbl, <span class="st">&quot;files/MAG/02_Bacteria_71.txt&quot;</span>, <span class="at">col.names  =</span> <span class="cn">TRUE</span>, </span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</details>
</div>
<div class="infobox">
<div class="center">
<p><i class="fas fa-file"></i> <strong>Data pack files</strong></p>
</div>
<p>02_Archaea_76.txt &amp; 02_Bacteria_71.txt</p>
</div>
<p>Sweet.</p>
<p>Now, we need to grab the list of all single copy genes from the <a href="https://gtdb.ecogenomic.org/">Genome Taxonomy Database</a> we used in assigning taxonomy to our collection of MAGs. One way to do this is to list the files in the <code>GTDB/SCG_SEARCH_DATABASES</code> directory within anvi’o. For that, a simple bash one-liner will do the trick. Alternatively, you can assess the same information on the <a href="https://github.com/meren/anvio/tree/master/anvio/data/misc/SCG_TAXONOMY/GTDB/SCG_SEARCH_DATABASES">anvi’o GitHub site</a>.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-1</span> PATH_TO_SCG_SEARCH_DATABASES <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&#39;s/.gz//&#39;</span></span></code></pre></div>
</div>
<div class="layout-chunk" data-layout="l-body">
<pre><code> [1] &quot;Ribosomal_L1&quot;   &quot;Ribosomal_L13&quot;  &quot;Ribosomal_L16&quot; 
 [4] &quot;Ribosomal_L17&quot;  &quot;Ribosomal_L2&quot;   &quot;Ribosomal_L20&quot; 
 [7] &quot;Ribosomal_L21p&quot; &quot;Ribosomal_L22&quot;  &quot;Ribosomal_L27A&quot;
[10] &quot;Ribosomal_L3&quot;   &quot;Ribosomal_L4&quot;   &quot;Ribosomal_L6&quot;  
[13] &quot;Ribosomal_L9_C&quot; &quot;Ribosomal_S11&quot;  &quot;Ribosomal_S2&quot;  
[16] &quot;Ribosomal_S20p&quot; &quot;Ribosomal_S3_C&quot; &quot;Ribosomal_S6&quot;  
[19] &quot;Ribosomal_S7&quot;   &quot;Ribosomal_S8&quot;   &quot;Ribosomal_S9&quot;  
[22] &quot;ribosomal_L24&quot; </code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="infobox">
<div class="center">
<p><i class="fas fa-file"></i> <strong>Data pack file</strong></p>
</div>
<p>03_GTDB_Taxonomy.txt</p>
</div>
<p>Now we create a master look up table containing all SCG from <code>Archaea_76</code> and <code>Bacteria_71</code>, then indicate which SCGs are also used in the GTDB_Taxonomy.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>arch_tbl<span class="sc">$</span>Archaea_76 <span class="ot">&lt;-</span> <span class="st">&quot;yes&quot;</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>bact_tbl<span class="sc">$</span>Bacteria_71 <span class="ot">&lt;-</span> <span class="st">&quot;yes&quot;</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>tmp_scg <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(arch_tbl, bact_tbl, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;genes&quot;</span>, <span class="st">&quot;accession&quot;</span>, <span class="st">&quot;pfam_link&quot;</span>), <span class="at">keep =</span> <span class="cn">FALSE</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>gtdb<span class="sc">$</span>GTDB_Taxonomy <span class="ot">&lt;-</span> <span class="st">&quot;yes&quot;</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>tmp_scg <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(tmp_scg, gtdb, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;genes&quot;</span> <span class="ot">=</span> <span class="st">&quot;genes&quot;</span>), <span class="at">keep =</span> <span class="cn">FALSE</span>)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>scg_table <span class="ot">&lt;-</span> tmp_scg</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(scg_table, <span class="st">&quot;files/MAG/04_SCG_master_list.txt&quot;</span>, <span class="at">col.names  =</span> <span class="cn">TRUE</span>, </span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(tmp_scg)</span></code></pre></div>
</details>
</div>
<p><small><strong>Table 2 | Concatonated list of all SCG in the Archaea_76 and Bacteria_71 HMM profiles. SCG used in the GTB taxonomy are also indicated. Note that Archaea_76 and Bacteria_71 share 25 SCGs, 12 of which are GTDB taxonomy genes.</strong></small></p>
<div class="layout-chunk" data-layout="l-body-outset">
<div id="htmlwidget-14e70aab3915a99b78fd" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-14e70aab3915a99b78fd">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"genes":["Adenylsucc_synt","AdoHcyase","Archease","ATP-synt_D","ATP-synt_F","CarS-like","CTP-dep_RFKase","Diphthamide_syn","DNA_primase_lrg","dsDNA_bind","DUF357","DUF359","DUF655","eIF-6","FbpA","Ham1p_like","HMG-CoA_red","NDK","PPS_PS","Prefoldin","PTH2","PyrI","Ribosomal_L1","Ribosomal_L13","Ribosomal_L14","Ribosomal_L15e","Ribosomal_L16","Ribosomal_L21e","Ribosomal_L22","Ribosomal_L23","Ribosomal_L26","Ribosomal_L29","Ribosomal_L3","Ribosomal_L31e","Ribosomal_L32e","Ribosomal_L37ae","Ribosomal_L39","Ribosomal_L4","Ribosomal_L44","Ribosomal_L5e","Ribosomal_L6","Ribosomal_S11","Ribosomal_S13","Ribosomal_S15","Ribosomal_S17","Ribosomal_S17e","Ribosomal_S19","Ribosomal_S19e","Ribosomal_S2","Ribosomal_S24e","Ribosomal_S27e","Ribosomal_S28e","Ribosomal_S3Ae","Ribosomal_S7","Ribosomal_S8","Ribosomal_S8e","Ribosomal_S9","Ribosom_S12_S23","Rib_5-P_isom_A","RNase_HII","RNA_pol_L_2","RNA_pol_N","RNA_pol_Rpb4","RNA_pol_Rpb6","RtcB","SecY","Spt4","TIM","Trm56","tRNA-synt_1c","tRNA-synt_1d","tRNA-synt_His","TruD","vATP-synt_AC39","vATP-synt_E","V_ATPase_I","ADK","AICARFT_IMPCHas","ATP-synt","ATP-synt_A","Chorismate_synt","EF_TS","eIF-1a","Exonuc_VII_L","GrpE","IPPT","OSCP","Pept_tRNA_hydro","PGK","RBFA","RecO_C","Ribonuclease_P","Ribosomal_L2","Ribosomal_L17","Ribosomal_L18p","Ribosomal_L19","Ribosomal_L20","Ribosomal_L21p","ribosomal_L24","Ribosomal_L27","Ribosomal_L27A","Ribosomal_L28","Ribosomal_L32p","Ribosomal_L35p","Ribosomal_L5","Ribosomal_L9_C","Ribosomal_S10","Ribosomal_S16","Ribosomal_S3_C","Ribosomal_S6","Ribosomal_S20p","RNA_pol_L","RRF","RsfS","RuvX","SecE","SecG","SmpB","tRNA_m1G_MT","TsaE","UPF0054","YajC"],"accession":["PF00709.21","PF05221.17","PF01951.16","PF01813.17","PF01990.17","PF01864.17","PF01982.16","PF01866.17","PF04104.14","PF01984.20","PF04010.13","PF04019.12","PF04919.12","PF01912.18","PF05833.11","PF01725.16","PF00368.18","PF00334.19","PF02006.16","PF02996.17","PF01981.16","PF01948.18","PF00687.21","PF00572.18","PF00238.19","PF00827.17","PF00252.18","PF01157.18","PF00237.19","PF00276.20","PF16906.5","PF00831.23","PF00297.22","PF01198.19","PF01655.18","PF01780.19","PF00832.20","PF00573.22","PF00935.19","PF17144.4","PF00347.23","PF00411.19","PF00416.22","PF00312.22","PF00366.20","PF00833.18","PF00203.21","PF01090.19","PF00318.20","PF01282.19","PF01667.17","PF01200.18","PF01015.18","PF00177.21","PF00410.19","PF01201.22","PF00380.19","PF00164.25","PF06026.14","PF01351.18","PF13656.6","PF01194.17","PF03874.16","PF01192.22","PF01139.17","PF00344.20","PF06093.13","PF00121.18","PF01994.16","PF00749.21","PF00750.19","PF13393.6","PF01142.18","PF01992.16","PF01991.18","PF01496.19","PF00406.22","PF01808.18","PF00231.19","PF00119.20","PF01264.21","PF00889.19","PF01176.19","PF02601.15","PF01025.19","PF01715.17","PF00213.18","PF01195.19","PF00162.19","PF02033.18","PF02565.15","PF00825.18","PF00181.23","PF01196.19","PF00861.22","PF01245.20","PF00453.18","PF00829.21","PF17136.4","PF01016.19","PF00828.19","PF00830.19","PF01783.23","PF01632.19","PF00281.19","PF03948.14","PF00338.22","PF00886.19","PF00189.20","PF01250.17","PF01649.18","PF01193.24","PF01765.19","PF02410.15","PF03652.15","PF00584.20","PF03840.14","PF01668.18","PF01746.21","PF02367.17","PF02130.17","PF02699.15"],"pfam_link":["https://pfam.xfam.org/family/PF00709.21","https://pfam.xfam.org/family/PF05221.17","https://pfam.xfam.org/family/PF01951.16","https://pfam.xfam.org/family/PF01813.17","https://pfam.xfam.org/family/PF01990.17","https://pfam.xfam.org/family/PF01864.17","https://pfam.xfam.org/family/PF01982.16","https://pfam.xfam.org/family/PF01866.17","https://pfam.xfam.org/family/PF04104.14","https://pfam.xfam.org/family/PF01984.20","https://pfam.xfam.org/family/PF04010.13","https://pfam.xfam.org/family/PF04019.12","https://pfam.xfam.org/family/PF04919.12","https://pfam.xfam.org/family/PF01912.18","https://pfam.xfam.org/family/PF05833.11","https://pfam.xfam.org/family/PF01725.16","https://pfam.xfam.org/family/PF00368.18","https://pfam.xfam.org/family/PF00334.19","https://pfam.xfam.org/family/PF02006.16","https://pfam.xfam.org/family/PF02996.17","https://pfam.xfam.org/family/PF01981.16","https://pfam.xfam.org/family/PF01948.18","https://pfam.xfam.org/family/PF00687.21","https://pfam.xfam.org/family/PF00572.18","https://pfam.xfam.org/family/PF00238.19","https://pfam.xfam.org/family/PF00827.17","https://pfam.xfam.org/family/PF00252.18","https://pfam.xfam.org/family/PF01157.18","https://pfam.xfam.org/family/PF00237.19","https://pfam.xfam.org/family/PF00276.20","https://pfam.xfam.org/family/PF16906.5","https://pfam.xfam.org/family/PF00831.23","https://pfam.xfam.org/family/PF00297.22","https://pfam.xfam.org/family/PF01198.19","https://pfam.xfam.org/family/PF01655.18","https://pfam.xfam.org/family/PF01780.19","https://pfam.xfam.org/family/PF00832.20","https://pfam.xfam.org/family/PF00573.22","https://pfam.xfam.org/family/PF00935.19","https://pfam.xfam.org/family/PF17144.4","https://pfam.xfam.org/family/PF00347.23","https://pfam.xfam.org/family/PF00411.19","https://pfam.xfam.org/family/PF00416.22","https://pfam.xfam.org/family/PF00312.22","https://pfam.xfam.org/family/PF00366.20","https://pfam.xfam.org/family/PF00833.18","https://pfam.xfam.org/family/PF00203.21","https://pfam.xfam.org/family/PF01090.19","https://pfam.xfam.org/family/PF00318.20","https://pfam.xfam.org/family/PF01282.19","https://pfam.xfam.org/family/PF01667.17","https://pfam.xfam.org/family/PF01200.18","https://pfam.xfam.org/family/PF01015.18","https://pfam.xfam.org/family/PF00177.21","https://pfam.xfam.org/family/PF00410.19","https://pfam.xfam.org/family/PF01201.22","https://pfam.xfam.org/family/PF00380.19","https://pfam.xfam.org/family/PF00164.25","https://pfam.xfam.org/family/PF06026.14","https://pfam.xfam.org/family/PF01351.18","https://pfam.xfam.org/family/PF13656.6","https://pfam.xfam.org/family/PF01194.17","https://pfam.xfam.org/family/PF03874.16","https://pfam.xfam.org/family/PF01192.22","https://pfam.xfam.org/family/PF01139.17","https://pfam.xfam.org/family/PF00344.20","https://pfam.xfam.org/family/PF06093.13","https://pfam.xfam.org/family/PF00121.18","https://pfam.xfam.org/family/PF01994.16","https://pfam.xfam.org/family/PF00749.21","https://pfam.xfam.org/family/PF00750.19","https://pfam.xfam.org/family/PF13393.6","https://pfam.xfam.org/family/PF01142.18","https://pfam.xfam.org/family/PF01992.16","https://pfam.xfam.org/family/PF01991.18","https://pfam.xfam.org/family/PF01496.19","https://pfam.xfam.org/family/PF00406.22","https://pfam.xfam.org/family/PF01808.18","https://pfam.xfam.org/family/PF00231.19","https://pfam.xfam.org/family/PF00119.20","https://pfam.xfam.org/family/PF01264.21","https://pfam.xfam.org/family/PF00889.19","https://pfam.xfam.org/family/PF01176.19","https://pfam.xfam.org/family/PF02601.15","https://pfam.xfam.org/family/PF01025.19","https://pfam.xfam.org/family/PF01715.17","https://pfam.xfam.org/family/PF00213.18","https://pfam.xfam.org/family/PF01195.19","https://pfam.xfam.org/family/PF00162.19","https://pfam.xfam.org/family/PF02033.18","https://pfam.xfam.org/family/PF02565.15","https://pfam.xfam.org/family/PF00825.18","https://pfam.xfam.org/family/PF00181.23","https://pfam.xfam.org/family/PF01196.19","https://pfam.xfam.org/family/PF00861.22","https://pfam.xfam.org/family/PF01245.20","https://pfam.xfam.org/family/PF00453.18","https://pfam.xfam.org/family/PF00829.21","https://pfam.xfam.org/family/PF17136.4","https://pfam.xfam.org/family/PF01016.19","https://pfam.xfam.org/family/PF00828.19","https://pfam.xfam.org/family/PF00830.19","https://pfam.xfam.org/family/PF01783.23","https://pfam.xfam.org/family/PF01632.19","https://pfam.xfam.org/family/PF00281.19","https://pfam.xfam.org/family/PF03948.14","https://pfam.xfam.org/family/PF00338.22","https://pfam.xfam.org/family/PF00886.19","https://pfam.xfam.org/family/PF00189.20","https://pfam.xfam.org/family/PF01250.17","https://pfam.xfam.org/family/PF01649.18","https://pfam.xfam.org/family/PF01193.24","https://pfam.xfam.org/family/PF01765.19","https://pfam.xfam.org/family/PF02410.15","https://pfam.xfam.org/family/PF03652.15","https://pfam.xfam.org/family/PF00584.20","https://pfam.xfam.org/family/PF03840.14","https://pfam.xfam.org/family/PF01668.18","https://pfam.xfam.org/family/PF01746.21","https://pfam.xfam.org/family/PF02367.17","https://pfam.xfam.org/family/PF02130.17","https://pfam.xfam.org/family/PF02699.15"],"Archaea_76":["yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],"Bacteria_71":["yes",null,null,null,null,null,null,null,null,null,null,null,null,null,null,"yes",null,null,null,null,null,null,"yes","yes","yes",null,"yes",null,"yes","yes",null,"yes","yes",null,null,null,null,"yes",null,null,"yes","yes","yes","yes","yes",null,"yes",null,"yes",null,null,null,null,"yes","yes",null,"yes","yes",null,null,null,null,null,"yes",null,"yes",null,null,null,null,"yes",null,null,null,null,null,"yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes"],"GTDB_Taxonomy":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"yes","yes",null,null,"yes",null,"yes",null,null,null,"yes",null,null,null,null,"yes",null,null,"yes","yes",null,null,null,null,null,null,"yes",null,null,null,null,"yes","yes",null,"yes",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"yes","yes",null,null,"yes","yes","yes",null,"yes",null,null,null,null,"yes",null,null,"yes","yes","yes",null,null,null,null,null,null,null,null,null,null,null]},"columns":[{"accessor":"genes","name":"genes","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["Adenylsucc_synt","AdoHcyase","Archease","ATP-synt_D","ATP-synt_F","CarS-like","CTP-dep_RFKase","Diphthamide_syn","DNA_primase_lrg","dsDNA_bind","DUF357","DUF359","DUF655","eIF-6","FbpA","Ham1p_like","HMG-CoA_red","NDK","PPS_PS","Prefoldin","PTH2","PyrI","Ribosomal_L1","Ribosomal_L13","Ribosomal_L14","Ribosomal_L15e","Ribosomal_L16","Ribosomal_L21e","Ribosomal_L22","Ribosomal_L23","Ribosomal_L26","Ribosomal_L29","Ribosomal_L3","Ribosomal_L31e","Ribosomal_L32e","Ribosomal_L37ae","Ribosomal_L39","Ribosomal_L4","Ribosomal_L44","Ribosomal_L5e","Ribosomal_L6","Ribosomal_S11","Ribosomal_S13","Ribosomal_S15","Ribosomal_S17","Ribosomal_S17e","Ribosomal_S19","Ribosomal_S19e","Ribosomal_S2","Ribosomal_S24e","Ribosomal_S27e","Ribosomal_S28e","Ribosomal_S3Ae","Ribosomal_S7","Ribosomal_S8","Ribosomal_S8e","Ribosomal_S9","Ribosom_S12_S23","Rib_5-P_isom_A","RNase_HII","RNA_pol_L_2","RNA_pol_N","RNA_pol_Rpb4","RNA_pol_Rpb6","RtcB","SecY","Spt4","TIM","Trm56","tRNA-synt_1c","tRNA-synt_1d","tRNA-synt_His","TruD","vATP-synt_AC39","vATP-synt_E","V_ATPase_I","ADK","AICARFT_IMPCHas","ATP-synt","ATP-synt_A","Chorismate_synt","EF_TS","eIF-1a","Exonuc_VII_L","GrpE","IPPT","OSCP","Pept_tRNA_hydro","PGK","RBFA","RecO_C","Ribonuclease_P","Ribosomal_L2","Ribosomal_L17","Ribosomal_L18p","Ribosomal_L19","Ribosomal_L20","Ribosomal_L21p","ribosomal_L24","Ribosomal_L27","Ribosomal_L27A","Ribosomal_L28","Ribosomal_L32p","Ribosomal_L35p","Ribosomal_L5","Ribosomal_L9_C","Ribosomal_S10","Ribosomal_S16","Ribosomal_S3_C","Ribosomal_S6","Ribosomal_S20p","RNA_pol_L","RRF","RsfS","RuvX","SecE","SecG","SmpB","tRNA_m1G_MT","TsaE","UPF0054","YajC"],"header":"genes","align":"left","footerStyle":{"fontWeight":"bold"},"minWidth":100,"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"accession","name":"accession","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["<a href=\"https://pfam.xfam.org/family/PF00709.21\" target=\"_blank\">PF00709.21<\/a>","<a href=\"https://pfam.xfam.org/family/PF05221.17\" target=\"_blank\">PF05221.17<\/a>","<a href=\"https://pfam.xfam.org/family/PF01951.16\" target=\"_blank\">PF01951.16<\/a>","<a href=\"https://pfam.xfam.org/family/PF01813.17\" target=\"_blank\">PF01813.17<\/a>","<a href=\"https://pfam.xfam.org/family/PF01990.17\" target=\"_blank\">PF01990.17<\/a>","<a href=\"https://pfam.xfam.org/family/PF01864.17\" target=\"_blank\">PF01864.17<\/a>","<a href=\"https://pfam.xfam.org/family/PF01982.16\" target=\"_blank\">PF01982.16<\/a>","<a href=\"https://pfam.xfam.org/family/PF01866.17\" target=\"_blank\">PF01866.17<\/a>","<a href=\"https://pfam.xfam.org/family/PF04104.14\" target=\"_blank\">PF04104.14<\/a>","<a href=\"https://pfam.xfam.org/family/PF01984.20\" target=\"_blank\">PF01984.20<\/a>","<a href=\"https://pfam.xfam.org/family/PF04010.13\" target=\"_blank\">PF04010.13<\/a>","<a href=\"https://pfam.xfam.org/family/PF04019.12\" target=\"_blank\">PF04019.12<\/a>","<a href=\"https://pfam.xfam.org/family/PF04919.12\" target=\"_blank\">PF04919.12<\/a>","<a href=\"https://pfam.xfam.org/family/PF01912.18\" target=\"_blank\">PF01912.18<\/a>","<a href=\"https://pfam.xfam.org/family/PF05833.11\" target=\"_blank\">PF05833.11<\/a>","<a href=\"https://pfam.xfam.org/family/PF01725.16\" target=\"_blank\">PF01725.16<\/a>","<a href=\"https://pfam.xfam.org/family/PF00368.18\" target=\"_blank\">PF00368.18<\/a>","<a href=\"https://pfam.xfam.org/family/PF00334.19\" target=\"_blank\">PF00334.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF02006.16\" target=\"_blank\">PF02006.16<\/a>","<a href=\"https://pfam.xfam.org/family/PF02996.17\" target=\"_blank\">PF02996.17<\/a>","<a href=\"https://pfam.xfam.org/family/PF01981.16\" target=\"_blank\">PF01981.16<\/a>","<a href=\"https://pfam.xfam.org/family/PF01948.18\" target=\"_blank\">PF01948.18<\/a>","<a href=\"https://pfam.xfam.org/family/PF00687.21\" target=\"_blank\">PF00687.21<\/a>","<a href=\"https://pfam.xfam.org/family/PF00572.18\" target=\"_blank\">PF00572.18<\/a>","<a href=\"https://pfam.xfam.org/family/PF00238.19\" target=\"_blank\">PF00238.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF00827.17\" target=\"_blank\">PF00827.17<\/a>","<a href=\"https://pfam.xfam.org/family/PF00252.18\" target=\"_blank\">PF00252.18<\/a>","<a href=\"https://pfam.xfam.org/family/PF01157.18\" target=\"_blank\">PF01157.18<\/a>","<a href=\"https://pfam.xfam.org/family/PF00237.19\" target=\"_blank\">PF00237.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF00276.20\" target=\"_blank\">PF00276.20<\/a>","<a href=\"https://pfam.xfam.org/family/PF16906.5\" target=\"_blank\">PF16906.5<\/a>","<a href=\"https://pfam.xfam.org/family/PF00831.23\" target=\"_blank\">PF00831.23<\/a>","<a href=\"https://pfam.xfam.org/family/PF00297.22\" target=\"_blank\">PF00297.22<\/a>","<a href=\"https://pfam.xfam.org/family/PF01198.19\" target=\"_blank\">PF01198.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF01655.18\" target=\"_blank\">PF01655.18<\/a>","<a href=\"https://pfam.xfam.org/family/PF01780.19\" target=\"_blank\">PF01780.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF00832.20\" target=\"_blank\">PF00832.20<\/a>","<a href=\"https://pfam.xfam.org/family/PF00573.22\" target=\"_blank\">PF00573.22<\/a>","<a href=\"https://pfam.xfam.org/family/PF00935.19\" target=\"_blank\">PF00935.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF17144.4\" target=\"_blank\">PF17144.4<\/a>","<a href=\"https://pfam.xfam.org/family/PF00347.23\" target=\"_blank\">PF00347.23<\/a>","<a href=\"https://pfam.xfam.org/family/PF00411.19\" target=\"_blank\">PF00411.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF00416.22\" target=\"_blank\">PF00416.22<\/a>","<a href=\"https://pfam.xfam.org/family/PF00312.22\" target=\"_blank\">PF00312.22<\/a>","<a href=\"https://pfam.xfam.org/family/PF00366.20\" target=\"_blank\">PF00366.20<\/a>","<a href=\"https://pfam.xfam.org/family/PF00833.18\" target=\"_blank\">PF00833.18<\/a>","<a href=\"https://pfam.xfam.org/family/PF00203.21\" target=\"_blank\">PF00203.21<\/a>","<a href=\"https://pfam.xfam.org/family/PF01090.19\" target=\"_blank\">PF01090.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF00318.20\" target=\"_blank\">PF00318.20<\/a>","<a href=\"https://pfam.xfam.org/family/PF01282.19\" target=\"_blank\">PF01282.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF01667.17\" target=\"_blank\">PF01667.17<\/a>","<a href=\"https://pfam.xfam.org/family/PF01200.18\" target=\"_blank\">PF01200.18<\/a>","<a href=\"https://pfam.xfam.org/family/PF01015.18\" target=\"_blank\">PF01015.18<\/a>","<a href=\"https://pfam.xfam.org/family/PF00177.21\" target=\"_blank\">PF00177.21<\/a>","<a href=\"https://pfam.xfam.org/family/PF00410.19\" target=\"_blank\">PF00410.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF01201.22\" target=\"_blank\">PF01201.22<\/a>","<a href=\"https://pfam.xfam.org/family/PF00380.19\" target=\"_blank\">PF00380.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF00164.25\" target=\"_blank\">PF00164.25<\/a>","<a href=\"https://pfam.xfam.org/family/PF06026.14\" target=\"_blank\">PF06026.14<\/a>","<a href=\"https://pfam.xfam.org/family/PF01351.18\" target=\"_blank\">PF01351.18<\/a>","<a href=\"https://pfam.xfam.org/family/PF13656.6\" target=\"_blank\">PF13656.6<\/a>","<a href=\"https://pfam.xfam.org/family/PF01194.17\" target=\"_blank\">PF01194.17<\/a>","<a href=\"https://pfam.xfam.org/family/PF03874.16\" target=\"_blank\">PF03874.16<\/a>","<a href=\"https://pfam.xfam.org/family/PF01192.22\" target=\"_blank\">PF01192.22<\/a>","<a href=\"https://pfam.xfam.org/family/PF01139.17\" target=\"_blank\">PF01139.17<\/a>","<a href=\"https://pfam.xfam.org/family/PF00344.20\" target=\"_blank\">PF00344.20<\/a>","<a href=\"https://pfam.xfam.org/family/PF06093.13\" target=\"_blank\">PF06093.13<\/a>","<a href=\"https://pfam.xfam.org/family/PF00121.18\" target=\"_blank\">PF00121.18<\/a>","<a href=\"https://pfam.xfam.org/family/PF01994.16\" target=\"_blank\">PF01994.16<\/a>","<a href=\"https://pfam.xfam.org/family/PF00749.21\" target=\"_blank\">PF00749.21<\/a>","<a href=\"https://pfam.xfam.org/family/PF00750.19\" target=\"_blank\">PF00750.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF13393.6\" target=\"_blank\">PF13393.6<\/a>","<a href=\"https://pfam.xfam.org/family/PF01142.18\" target=\"_blank\">PF01142.18<\/a>","<a href=\"https://pfam.xfam.org/family/PF01992.16\" target=\"_blank\">PF01992.16<\/a>","<a href=\"https://pfam.xfam.org/family/PF01991.18\" target=\"_blank\">PF01991.18<\/a>","<a href=\"https://pfam.xfam.org/family/PF01496.19\" target=\"_blank\">PF01496.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF00406.22\" target=\"_blank\">PF00406.22<\/a>","<a href=\"https://pfam.xfam.org/family/PF01808.18\" target=\"_blank\">PF01808.18<\/a>","<a href=\"https://pfam.xfam.org/family/PF00231.19\" target=\"_blank\">PF00231.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF00119.20\" target=\"_blank\">PF00119.20<\/a>","<a href=\"https://pfam.xfam.org/family/PF01264.21\" target=\"_blank\">PF01264.21<\/a>","<a href=\"https://pfam.xfam.org/family/PF00889.19\" target=\"_blank\">PF00889.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF01176.19\" target=\"_blank\">PF01176.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF02601.15\" target=\"_blank\">PF02601.15<\/a>","<a href=\"https://pfam.xfam.org/family/PF01025.19\" target=\"_blank\">PF01025.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF01715.17\" target=\"_blank\">PF01715.17<\/a>","<a href=\"https://pfam.xfam.org/family/PF00213.18\" target=\"_blank\">PF00213.18<\/a>","<a href=\"https://pfam.xfam.org/family/PF01195.19\" target=\"_blank\">PF01195.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF00162.19\" target=\"_blank\">PF00162.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF02033.18\" target=\"_blank\">PF02033.18<\/a>","<a href=\"https://pfam.xfam.org/family/PF02565.15\" target=\"_blank\">PF02565.15<\/a>","<a href=\"https://pfam.xfam.org/family/PF00825.18\" target=\"_blank\">PF00825.18<\/a>","<a href=\"https://pfam.xfam.org/family/PF00181.23\" target=\"_blank\">PF00181.23<\/a>","<a href=\"https://pfam.xfam.org/family/PF01196.19\" target=\"_blank\">PF01196.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF00861.22\" target=\"_blank\">PF00861.22<\/a>","<a href=\"https://pfam.xfam.org/family/PF01245.20\" target=\"_blank\">PF01245.20<\/a>","<a href=\"https://pfam.xfam.org/family/PF00453.18\" target=\"_blank\">PF00453.18<\/a>","<a href=\"https://pfam.xfam.org/family/PF00829.21\" target=\"_blank\">PF00829.21<\/a>","<a href=\"https://pfam.xfam.org/family/PF17136.4\" target=\"_blank\">PF17136.4<\/a>","<a href=\"https://pfam.xfam.org/family/PF01016.19\" target=\"_blank\">PF01016.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF00828.19\" target=\"_blank\">PF00828.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF00830.19\" target=\"_blank\">PF00830.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF01783.23\" target=\"_blank\">PF01783.23<\/a>","<a href=\"https://pfam.xfam.org/family/PF01632.19\" target=\"_blank\">PF01632.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF00281.19\" target=\"_blank\">PF00281.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF03948.14\" target=\"_blank\">PF03948.14<\/a>","<a href=\"https://pfam.xfam.org/family/PF00338.22\" target=\"_blank\">PF00338.22<\/a>","<a href=\"https://pfam.xfam.org/family/PF00886.19\" target=\"_blank\">PF00886.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF00189.20\" target=\"_blank\">PF00189.20<\/a>","<a href=\"https://pfam.xfam.org/family/PF01250.17\" target=\"_blank\">PF01250.17<\/a>","<a href=\"https://pfam.xfam.org/family/PF01649.18\" target=\"_blank\">PF01649.18<\/a>","<a href=\"https://pfam.xfam.org/family/PF01193.24\" target=\"_blank\">PF01193.24<\/a>","<a href=\"https://pfam.xfam.org/family/PF01765.19\" target=\"_blank\">PF01765.19<\/a>","<a href=\"https://pfam.xfam.org/family/PF02410.15\" target=\"_blank\">PF02410.15<\/a>","<a href=\"https://pfam.xfam.org/family/PF03652.15\" target=\"_blank\">PF03652.15<\/a>","<a href=\"https://pfam.xfam.org/family/PF00584.20\" target=\"_blank\">PF00584.20<\/a>","<a href=\"https://pfam.xfam.org/family/PF03840.14\" target=\"_blank\">PF03840.14<\/a>","<a href=\"https://pfam.xfam.org/family/PF01668.18\" target=\"_blank\">PF01668.18<\/a>","<a href=\"https://pfam.xfam.org/family/PF01746.21\" target=\"_blank\">PF01746.21<\/a>","<a href=\"https://pfam.xfam.org/family/PF02367.17\" target=\"_blank\">PF02367.17<\/a>","<a href=\"https://pfam.xfam.org/family/PF02130.17\" target=\"_blank\">PF02130.17<\/a>","<a href=\"https://pfam.xfam.org/family/PF02699.15\" target=\"_blank\">PF02699.15<\/a>"],"header":"accession","align":"left","footerStyle":{"fontWeight":"bold"},"html":true,"minWidth":100},{"accessor":"pfam_link","name":"pfam_link","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":[{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00709.21","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00709.21"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF05221.17","target":"_blank"},"children":["https://pfam.xfam.org/family/PF05221.17"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01951.16","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01951.16"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01813.17","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01813.17"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01990.17","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01990.17"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01864.17","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01864.17"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01982.16","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01982.16"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01866.17","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01866.17"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF04104.14","target":"_blank"},"children":["https://pfam.xfam.org/family/PF04104.14"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01984.20","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01984.20"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF04010.13","target":"_blank"},"children":["https://pfam.xfam.org/family/PF04010.13"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF04019.12","target":"_blank"},"children":["https://pfam.xfam.org/family/PF04019.12"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF04919.12","target":"_blank"},"children":["https://pfam.xfam.org/family/PF04919.12"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01912.18","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01912.18"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF05833.11","target":"_blank"},"children":["https://pfam.xfam.org/family/PF05833.11"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01725.16","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01725.16"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00368.18","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00368.18"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00334.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00334.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF02006.16","target":"_blank"},"children":["https://pfam.xfam.org/family/PF02006.16"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF02996.17","target":"_blank"},"children":["https://pfam.xfam.org/family/PF02996.17"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01981.16","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01981.16"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01948.18","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01948.18"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00687.21","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00687.21"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00572.18","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00572.18"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00238.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00238.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00827.17","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00827.17"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00252.18","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00252.18"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01157.18","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01157.18"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00237.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00237.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00276.20","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00276.20"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF16906.5","target":"_blank"},"children":["https://pfam.xfam.org/family/PF16906.5"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00831.23","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00831.23"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00297.22","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00297.22"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01198.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01198.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01655.18","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01655.18"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01780.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01780.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00832.20","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00832.20"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00573.22","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00573.22"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00935.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00935.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF17144.4","target":"_blank"},"children":["https://pfam.xfam.org/family/PF17144.4"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00347.23","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00347.23"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00411.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00411.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00416.22","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00416.22"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00312.22","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00312.22"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00366.20","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00366.20"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00833.18","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00833.18"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00203.21","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00203.21"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01090.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01090.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00318.20","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00318.20"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01282.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01282.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01667.17","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01667.17"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01200.18","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01200.18"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01015.18","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01015.18"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00177.21","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00177.21"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00410.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00410.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01201.22","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01201.22"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00380.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00380.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00164.25","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00164.25"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF06026.14","target":"_blank"},"children":["https://pfam.xfam.org/family/PF06026.14"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01351.18","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01351.18"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF13656.6","target":"_blank"},"children":["https://pfam.xfam.org/family/PF13656.6"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01194.17","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01194.17"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF03874.16","target":"_blank"},"children":["https://pfam.xfam.org/family/PF03874.16"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01192.22","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01192.22"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01139.17","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01139.17"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00344.20","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00344.20"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF06093.13","target":"_blank"},"children":["https://pfam.xfam.org/family/PF06093.13"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00121.18","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00121.18"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01994.16","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01994.16"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00749.21","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00749.21"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00750.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00750.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF13393.6","target":"_blank"},"children":["https://pfam.xfam.org/family/PF13393.6"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01142.18","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01142.18"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01992.16","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01992.16"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01991.18","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01991.18"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01496.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01496.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00406.22","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00406.22"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01808.18","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01808.18"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00231.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00231.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00119.20","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00119.20"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01264.21","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01264.21"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00889.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00889.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01176.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01176.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF02601.15","target":"_blank"},"children":["https://pfam.xfam.org/family/PF02601.15"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01025.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01025.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01715.17","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01715.17"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00213.18","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00213.18"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01195.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01195.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00162.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00162.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF02033.18","target":"_blank"},"children":["https://pfam.xfam.org/family/PF02033.18"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF02565.15","target":"_blank"},"children":["https://pfam.xfam.org/family/PF02565.15"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00825.18","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00825.18"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00181.23","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00181.23"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01196.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01196.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00861.22","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00861.22"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01245.20","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01245.20"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00453.18","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00453.18"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00829.21","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00829.21"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF17136.4","target":"_blank"},"children":["https://pfam.xfam.org/family/PF17136.4"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01016.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01016.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00828.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00828.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00830.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00830.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01783.23","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01783.23"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01632.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01632.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00281.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00281.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF03948.14","target":"_blank"},"children":["https://pfam.xfam.org/family/PF03948.14"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00338.22","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00338.22"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00886.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00886.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00189.20","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00189.20"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01250.17","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01250.17"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01649.18","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01649.18"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01193.24","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01193.24"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01765.19","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01765.19"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF02410.15","target":"_blank"},"children":["https://pfam.xfam.org/family/PF02410.15"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF03652.15","target":"_blank"},"children":["https://pfam.xfam.org/family/PF03652.15"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF00584.20","target":"_blank"},"children":["https://pfam.xfam.org/family/PF00584.20"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF03840.14","target":"_blank"},"children":["https://pfam.xfam.org/family/PF03840.14"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01668.18","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01668.18"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF01746.21","target":"_blank"},"children":["https://pfam.xfam.org/family/PF01746.21"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF02367.17","target":"_blank"},"children":["https://pfam.xfam.org/family/PF02367.17"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF02130.17","target":"_blank"},"children":["https://pfam.xfam.org/family/PF02130.17"]},{"name":"a","attribs":{"href":"https://pfam.xfam.org/family/PF02699.15","target":"_blank"},"children":["https://pfam.xfam.org/family/PF02699.15"]}],"header":"pfam link","align":"left","footerStyle":{"fontWeight":"bold"},"show":false},{"accessor":"Archaea_76","name":"Archaea_76","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA"],"header":"Archaea 76","align":"left","footerStyle":{"fontWeight":"bold"}},{"accessor":"Bacteria_71","name":"Bacteria_71","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["yes","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","yes","NA","NA","NA","NA","NA","NA","yes","yes","yes","NA","yes","NA","yes","yes","NA","yes","yes","NA","NA","NA","NA","yes","NA","NA","yes","yes","yes","yes","yes","NA","yes","NA","yes","NA","NA","NA","NA","yes","yes","NA","yes","yes","NA","NA","NA","NA","NA","yes","NA","yes","NA","NA","NA","NA","yes","NA","NA","NA","NA","NA","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes","yes"],"header":"Bacteria 71","align":"left","footerStyle":{"fontWeight":"bold"}},{"accessor":"GTDB_Taxonomy","name":"GTDB_Taxonomy","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","yes","yes","NA","NA","yes","NA","yes","NA","NA","NA","yes","NA","NA","NA","NA","yes","NA","NA","yes","yes","NA","NA","NA","NA","NA","NA","yes","NA","NA","NA","NA","yes","yes","NA","yes","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","yes","yes","NA","NA","yes","yes","yes","NA","yes","NA","NA","NA","NA","yes","NA","NA","yes","yes","yes","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA"],"header":"GTDB Taxonomy","align":"left","footerStyle":{"fontWeight":"bold"}}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,122],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"b500fb0605e3be7ca8fc6a3fb9c0a6ca"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
<div class="infobox">
<div class="center">
<p><i class="fas fa-file"></i> <strong>Data pack file</strong></p>
</div>
<p>04_SCG_master_list.txt</a></p>
</div>
<blockquote>
<p>It is important to note that <code>Archaea_76</code> and <code>Bacteria_71</code> share <strong>25</strong> SCGs, 12 of which are GTDB taxonomy genes.</p>
</blockquote>
<h2 id="mag-completeness-taxonomy">MAG Completeness &amp; Taxonomy</h2>
<p>Now we can generate some stats about our MAG collection. We define a MAG as:</p>
<ul>
<li>Greater than 70% completion AND less than 10% redundancy</li>
</ul>
<p>OR</p>
<ul>
<li>Larger than 1Mb genome size AND less than 10% redundancy</li>
</ul>
<p>What this means is that even if a bin has &lt; 70% completion, we call it a MAG as long as it is &gt; 1Mb in size. MAGs are denoted by the prefix <code>WA_MAG</code>. We also included 9 bins in this collection that had 10 or more total GTDB SCGs, despite not meeting either of the criteria described above. These are denoted by the prefix <code>WA_Bin</code>.</p>
<p>First we estimate genome completeness.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-estimate-genome-completeness</span> <span class="at">--contigs-db</span> 03_CONTIGS/WA-contigs.db <span class="dt">\</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">--profile-db</span> 06_MERGED/WA/PROFILE.db <span class="dt">\</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">--collection-name</span> DASTOOL_MAGS <span class="dt">\</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">--output-file</span> 05_MAG-COMPLETENESS.txt</span></code></pre></div>
<div class="infobox">
<div class="center">
<p><i class="fas fa-file"></i> <strong>Data pack file</strong></p>
</div>
<p>05_MAG-COMPLETENESS.txt</a></p>
</div>
<p>Then let’s estimate MAG consensus taxonomy using <code>anvi-estimate-scg-taxonomy</code>. Here we add the flag <code>--compute-scg-coverages</code> to calculate coverage statistics of single-copy core genes for which we have taxonomic information. <strong>Note this is not the same as computing coverage statistics for the entire MAG</strong>, which we will do later in the workflow.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-estimate-scg-taxonomy</span> <span class="at">--profile-db</span> 06_MERGED/WA/PROFILE.db <span class="dt">\</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">--contigs-db</span> 03_CONTIGS/WA-contigs.db  <span class="dt">\</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">--output-file</span> 05_MAG-TAXONOMY.txt  <span class="dt">\</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">--compute-scg-coverages</span>  <span class="dt">\</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">--collection-name</span> DASTOOL_MAGS <span class="dt">\</span></span></code></pre></div>
<div class="infobox">
<div class="center">
<p><i class="fas fa-file"></i> <strong>Data pack file</strong></p>
</div>
<p>05_MAG-TAXONOMY.txt</a></p>
</div>
<p>We can also rerun the command using the flag <code>--per-scg-output-file</code>, which provides taxonomic information and percent identity scores for each SCG in a MAG. So rather than a consensus taxonomy for each MAG, we get info for each SCG hit in a MAG.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-estimate-scg-taxonomy</span> <span class="at">--profile-db</span> 06_MERGED/WA/PROFILE.db <span class="dt">\</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">--contigs-db</span> 03_CONTIGS/WA-contigs.db  <span class="dt">\</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">--collection-name</span> DASTOOL_MAGS <span class="dt">\</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">--per-scg-output-file</span> 05_MAG-TAXONOMY-PER-GENOME-SCG.txt</span></code></pre></div>
<div class="infobox">
<div class="center">
<p><i class="fas fa-file"></i> <strong>Data pack file</strong></p>
</div>
<p>05_MAG-TAXONOMY-PER-GENOME-SCG.txt</a></p>
</div>
<p>OK, now we can merge the MAG completeness and the MAG taxonomy tables and view the data. We will omit the coverage statistics for now so we can focus on the MAG details.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><small><strong>Table 3 | Completeness and taxonomic estimates of all bins in the MAG collection. A bin is considered a MAG if it is either greater than 70% completion AND less than 10% redundancy OR larger than 1Mb genome size AND less than 10% redundancy. MAGs are denoted by the prefix WA_MAG. 9 bins additional bins are included that have 10 or more total GTDB SCGs, despite not meeting either of the criteria described above. These are denoted by the prefix WA_Bin.</strong></small></p>
<div class="layout-chunk" data-layout="l-page">
<div id="htmlwidget-0a8bc5ea074c2f778b98" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-0a8bc5ea074c2f778b98">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"bin name":["WA_MAG_00001","WA_MAG_00002","WA_MAG_00003","WA_MAG_00004","WA_MAG_00005","WA_MAG_00006","WA_MAG_00007","WA_MAG_00008","WA_MAG_00009","WA_MAG_00010","WA_MAG_00011","WA_MAG_00012","WA_MAG_00013","WA_MAG_00014","WA_MAG_00015","WA_MAG_00016","WA_MAG_00017","WA_MAG_00018","WA_MAG_00019","WA_MAG_00020","WA_MAG_00021","WA_MAG_00022","WA_MAG_00023","WA_MAG_00024","WA_MAG_00025","WA_MAG_00026","WA_MAG_00027","WA_MAG_00028","WA_MAG_00029","WA_MAG_00030","WA_MAG_00031","WA_MAG_00032","WA_MAG_00033","WA_MAG_00034","WA_MAG_00035","WA_MAG_00036","WA_MAG_00037","WA_MAG_00038","WA_MAG_00039","WA_MAG_00040","WA_MAG_00041","WA_MAG_00042","WA_MAG_00043","WA_MAG_00044","WA_MAG_00045","WA_MAG_00046","WA_MAG_00047","WA_MAG_00048","WA_MAG_00049","WA_MAG_00050","WA_MAG_00051","WA_MAG_00052","WA_MAG_00053","WA_MAG_00054","WA_MAG_00055","WA_MAG_00056","WA_MAG_00057","WA_MAG_00058","WA_MAG_00059","WA_MAG_00060","WA_MAG_00061","WA_MAG_00062","WA_MAG_00063","WA_MAG_00064","WA_MAG_00065","WA_MAG_00066","WA_MAG_00067","WA_MAG_00068","WA_MAG_00069","WA_MAG_00070","WA_MAG_00071","WA_MAG_00072","WA_MAG_00073","WA_MAG_00074","WA_MAG_00075","WA_MAG_00076","WA_MAG_00077","WA_MAG_00078","WA_MAG_00079","WA_MAG_00080","WA_MAG_00081","WA_MAG_00082","WA_MAG_00083","WA_MAG_00084","WA_MAG_00085","WA_MAG_00086","WA_MAG_00087","WA_MAG_00088","WA_MAG_00089","WA_Bin_00090","WA_Bin_00091","WA_Bin_00093","WA_Bin_00094","WA_Bin_00095","WA_Bin_00097","WA_Bin_00098","WA_Bin_00099","WA_Bin_00105"],"domain":["BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","EUKARYA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","ARCHAEA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","ARCHAEA","ARCHAEA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","BACTERIA","ARCHAEA","ARCHAEA","BACTERIA","ARCHAEA","ARCHAEA","BACTERIA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","BACTERIA","ARCHAEA","BACTERIA","BACTERIA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA"],"confidence":[1,1,1,1,1,1,1,1,1,0.6,1,1,1,1,1,0.9,1,1,1,1,1,1,0.8,0.9,0.9,0.9,0.9,0.9,0.9,0.8,1,0.9,0.9,0.9,0.9,1,0.9,0.9,0.9,0.9,0.8,0.9,1,0.8,0.7,0.8,0.8,0.8,0.8,0.8,0.9,0.9,0.9,1,1,0.9,0.9,0.9,0.8,0.8,0.8,0.8,0.8,0.8,0.6,0.8,0.7,0.7,0.8,0.7,0.7,0.8,0.7,0.8,0.7,0.7,0.7,0.4,0.6,0.6,0.6,0.7,0.5,0.4,0.6,0.2,0.2,0.2,0.2,0.7,0.6,0.7,0.7,0.7,0.5,0.5,0.4,0.3],"% completion":[100,100,98.59,97.18,97.18,97.18,98.59,98.59,95.77,96.39,95.77,95.77,94.37,97.18,97.18,97.18,94.37,92.96,97.18,91.55,98.59,92.96,91.55,91.55,94.37,95.77,92.96,90.14,88.16,88.16,92.11,94.37,90.14,88.73,90.14,90.14,88.73,84.51,84.51,88.73,90.79,85.92,87.32,83.1,82.89,84.21,81.58,84.21,80.28,81.69,85.92,84.51,84.51,85.92,83.1,78.87,84.51,84.51,80.26,83.1,80.26,78.95,81.69,80.26,72.37,77.46,72.37,73.24,73.24,70.42,70.42,69.01,66.2,71.83,71.83,67.11,67.61,59.21,64.79,66.2,63.16,54.93,53.52,50.7,50.7,29.58,29.58,30.99,28.17,66.2,63.38,64.79,66.2,57.75,52.11,52.11,35.21,32.39],"% redundancy":[1.41,1.41,0,0,0,0,2.82,2.82,0,1.2,1.41,1.41,0,4.23,4.23,4.23,1.41,0,5.63,0,7.04,1.41,1.41,2.82,5.63,8.45,5.63,2.82,1.32,1.32,5.26,8.45,4.23,4.23,5.63,5.63,5.63,1.41,1.41,5.63,7.89,4.23,5.63,2.82,2.63,5.26,2.63,5.26,1.41,2.82,7.04,5.63,5.63,8.45,5.63,1.41,8.45,8.45,5.26,8.45,6.58,5.26,8.45,7.89,0,5.63,1.32,4.23,4.23,2.82,4.23,2.82,1.41,7.04,7.04,3.95,7.04,0,5.63,8.45,6.58,0,1.41,1.41,1.41,0,2.82,5.63,2.82,0,1.41,2.82,5.63,5.63,1.41,2.82,0,7.04],"num_splits":[137,93,66,224,101,143,310,165,97,840,662,127,191,447,258,174,419,145,728,161,105,146,365,279,495,568,353,823,116,65,99,1134,853,168,179,628,1007,243,110,630,435,201,897,412,159,164,628,166,348,321,669,909,1517,701,125,469,301,196,120,105,479,521,862,388,147,883,134,228,414,523,1413,287,636,298,443,96,280,251,581,737,393,1061,645,187,361,497,634,488,581,91,43,126,138,119,126,296,123,116],"total length":[2047586,1955749,1181014,2120415,1846770,1777287,1975778,2144605,1784192,12083237,3962024,1610000,2679653,2281835,1793838,1856990,2860427,2124535,1284585,1868016,1946327,1835621,2030837,2717342,1225026,1775013,2802924,1941485,1304622,1285593,1420955,2546495,4276881,2483765,1563428,2021430,3118345,1671594,682107,1823682,1924227,1291339,3311819,919638,1409520,1298967,1532470,1887979,2650093,1885399,2824715,2920759,2765431,1624286,672532,2043619,1957807,1487711,1455552,862504,1601316,2115568,2938562,1869954,1338986,1358926,1497594,1175570,2282132,1370180,3505296,1421650,1424896,1483903,1668075,1135864,1991795,1123467,1279612,1703261,1999441,1903263,1153401,1887118,1261026,1045899,1266422,1427867,1065789,940459,318414,701568,673603,562070,710238,660850,948063,179192],"total_scgs":[20,20,10,22,16,21,21,22,16,0,22,19,22,8,22,21,20,22,22,20,21,21,21,20,20,9,20,17,7,10,10,17,17,21,22,22,21,19,1,20,7,24,23,4,7,9,0,9,16,10,19,22,0,21,21,16,20,19,8,13,6,8,19,7,8,20,4,18,14,12,17,16,0,19,17,5,14,7,13,15,7,0,12,9,11,4,5,4,4,19,17,19,15,17,14,15,10,15],"supporting_scgs":[19,16,6,19,16,19,16,8,15,0,22,19,22,4,11,12,12,22,22,20,15,13,17,8,20,3,16,11,6,9,6,15,15,16,10,18,12,19,1,19,7,14,23,4,7,9,0,8,16,5,8,16,0,21,21,9,9,12,5,11,6,6,17,4,7,8,4,16,13,12,17,13,0,11,14,5,7,4,13,11,7,0,11,4,11,2,3,4,2,17,17,9,14,17,11,7,5,15],"t_domain":["Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","None","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Archaea","Archaea","Archaea","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Archaea","Bacteria","Bacteria","Bacteria","Archaea","Archaea","None","Archaea","Bacteria","Bacteria","Bacteria","Bacteria","None","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Archaea","Bacteria","Archaea","Archaea","Bacteria","Archaea","Archaea","Bacteria","Archaea","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","None","Bacteria","Bacteria","Archaea","Bacteria","Archaea","Bacteria","Bacteria","Archaea","None","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria"],"t_phylum":["Proteobacteria","Proteobacteria","Proteobacteria","Bacteroidota","Bacteroidota","Bacteroidota","Bacteroidota","Proteobacteria","Verrucomicrobiota","None","Verrucomicrobiota","Actinobacteriota","Proteobacteria","Proteobacteria","Bacteroidota","Proteobacteria","Planctomycetota","Actinobacteriota","Proteobacteria","Proteobacteria","Bacteroidota","Actinobacteriota","Proteobacteria","Proteobacteria","Proteobacteria","Proteobacteria","Actinobacteriota","Actinobacteriota","Thermoplasmatota","Thermoplasmatota","Thermoplasmatota","Proteobacteria","Planctomycetota","Proteobacteria","Bacteroidota","Bacteroidota","Planctomycetota","Bacteroidota","Firmicutes","Cyanobacteria","Thermoplasmatota","Bacteroidota","SAR324","Campylobacterota","Asgardarchaeota","Thermoplasmatota","None","Thermoplasmatota","Proteobacteria","Bacteroidota","Actinobacteriota","SAR324","None","Bacteroidota","Marinisomatota","Proteobacteria","Proteobacteria","Proteobacteria","Thermoplasmatota","Actinobacteriota","Thermoplasmatota","Thermoplasmatota","Verrucomicrobiota","Thermoplasmatota","Thermoplasmatota","Planctomycetota","Thermoplasmatota","Bacteroidota","Verrucomicrobiota","Bacteroidota","Planctomycetota","Proteobacteria","None","Proteobacteria","Bacteroidota","Thermoplasmatota","Proteobacteria","Thermoplasmatota","Proteobacteria","Bacteroidota","Thermoplasmatota","None","Proteobacteria","Bacteroidota","Cyanobacteria","Bacteroidota","Bacteroidota","Bacteroidota","Proteobacteria","Bacteroidota","Actinobacteriota","Bacteroidota","Proteobacteria","Proteobacteria","Proteobacteria","Bacteroidota","Proteobacteria","Bacteroidota"],"t_class":["Alphaproteobacteria","Alphaproteobacteria","Gammaproteobacteria","Bacteroidia","Bacteroidia","Bacteroidia","Bacteroidia","Alphaproteobacteria","Verrucomicrobiae","None","Verrucomicrobiae","Acidimicrobiia","Gammaproteobacteria","Alphaproteobacteria","Bacteroidia","Alphaproteobacteria","Planctomycetes","Acidimicrobiia","Gammaproteobacteria","Gammaproteobacteria","Bacteroidia","Acidimicrobiia","Gammaproteobacteria","Alphaproteobacteria","Gammaproteobacteria","Alphaproteobacteria","Acidimicrobiia","Acidimicrobiia","Poseidoniia","Poseidoniia","Poseidoniia","Alphaproteobacteria","Planctomycetes","Gammaproteobacteria","Bacteroidia","Bacteroidia","UBA8108","Bacteroidia","Bacilli","Cyanobacteriia","Poseidoniia","Bacteroidia","SAR324","Campylobacteria","Heimdallarchaeia","Poseidoniia","None","Poseidoniia","Alphaproteobacteria","Bacteroidia","Acidimicrobiia","SAR324","None","Bacteroidia","Marinisomatia","Alphaproteobacteria","Gammaproteobacteria","Alphaproteobacteria","Poseidoniia","Acidimicrobiia","Poseidoniia","Poseidoniia","Verrucomicrobiae","Poseidoniia","Poseidoniia","Phycisphaerae","Poseidoniia","Bacteroidia","Verrucomicrobiae","Bacteroidia","Planctomycetes","Gammaproteobacteria","None","Gammaproteobacteria","Bacteroidia","Poseidoniia","Alphaproteobacteria","Poseidoniia","Gammaproteobacteria","Bacteroidia","Poseidoniia","None","Gammaproteobacteria","Bacteroidia","Cyanobacteriia","Bacteroidia","Bacteroidia","Bacteroidia","Gammaproteobacteria","Bacteroidia","Acidimicrobiia","Bacteroidia","Gammaproteobacteria","Gammaproteobacteria","Alphaproteobacteria","Bacteroidia","Alphaproteobacteria","Bacteroidia"],"t_order":["Puniceispirillales","Puniceispirillales","PS1","Flavobacteriales","Flavobacteriales","Flavobacteriales","Flavobacteriales","Puniceispirillales","Opitutales","None","Pedosphaerales","Acidimicrobiales","Pseudomonadales","Puniceispirillales","Flavobacteriales","Puniceispirillales","Pirellulales","Acidimicrobiales","PS1","Pseudomonadales","Flavobacteriales","Acidimicrobiales","Pseudomonadales","Rhodobacterales","Chromatiales","Puniceispirillales","Acidimicrobiales","Acidimicrobiales","MGIII","Poseidoniales","Poseidoniales","Rhodobacterales","Pirellulales","Pseudomonadales","Cytophagales","Flavobacteriales","UBA1146","Flavobacteriales","Acholeplasmatales","PCC-6307","Poseidoniales","Flavobacteriales","SAR324","Campylobacterales","LC-2","Poseidoniales","None","Poseidoniales","Rhodobacterales","Flavobacteriales","Acidimicrobiales","SAR324","None","Flavobacteriales","Marinisomatales","Puniceispirillales","Enterobacterales","Rhodobacterales","Poseidoniales","Actinomarinales","Poseidoniales","Poseidoniales","Pedosphaerales","Poseidoniales","Poseidoniales","Phycisphaerales","Poseidoniales","Flavobacteriales","Verrucomicrobiales","Flavobacteriales","Pirellulales","Pseudomonadales","None","UBA11654","Flavobacteriales","Poseidoniales","Rhodobacterales","Poseidoniales","Pseudomonadales","Flavobacteriales","Poseidoniales","None","Methylococcales","Flavobacteriales","PCC-6307","Flavobacteriales","Flavobacteriales","Flavobacteriales","Chromatiales","Flavobacteriales","Actinomarinales","Flavobacteriales","SAR86","Pseudomonadales","Thalassobaculales","Flavobacteriales","Rhodobacterales","Flavobacteriales"],"t_family":["Puniceispirillaceae","Puniceispirillaceae","Thioglobaceae","Flavobacteriaceae","UA16","Flavobacteriaceae","Flavobacteriaceae","Puniceispirillaceae","DSM-45221","None","AAA164-E04","MedAcidi-G1","Pseudohongiellaceae","UBA1172","Flavobacteriaceae","Puniceispirillaceae","UBA1268","Ilumatobacteraceae","Thioglobaceae","Litoricolaceae","Flavobacteriaceae","UBA8592","Nitrincolaceae","Rhodobacteraceae","Sedimenticolaceae","Puniceispirillaceae","UBA11606","Ilumatobacteraceae","CG-Epi1","Thalassarchaeaceae","Thalassarchaeaceae","Rhodobacteraceae","Pirellulaceae","Nitrincolaceae","Marinoscillaceae","Schleiferiaceae","UBA1146","Schleiferiaceae","Acholeplasmataceae","Cyanobiaceae","Poseidoniaceae","Flavobacteriaceae","NAC60-12","Arcobacteraceae","LC-2","Thalassarchaeaceae","None","Poseidoniaceae","Rhodobacteraceae","Flavobacteriaceae","UBA11606","NAC60-12","None","Flavobacteriaceae","Marinisomataceae","Puniceispirillaceae","Alteromonadaceae","Rhodobacteraceae","Poseidoniaceae","Actinomarinaceae","Poseidoniaceae","Poseidoniaceae","UBA1100","Poseidoniaceae","Poseidoniaceae","SM1A02","Poseidoniaceae","Flavobacteriaceae","DEV007","UA16","Pirellulaceae","Litoricolaceae","None","UBA11654","TMED113","Thalassarchaeaceae","Rhodobacteraceae","Thalassarchaeaceae","Litoricolaceae","Crocinitomicaceae","Poseidoniaceae","None","Cycloclasticaceae","UA16","Cyanobiaceae","Flavobacteriaceae","UA16","Schleiferiaceae","Sedimenticolaceae","Flavobacteriaceae","Actinomarinaceae","BACL11","D2472","Halieaceae","Nisaeaceae","Flavobacteriaceae","Rhodobacteraceae","UA16"],"t_genus":["WTJO01","UBA685","Thiodubiliella","UBA3478","UBA974","RFYP01","BACL21","UBA8309","Coraliomargarita","None","UBA7971","MedAcidi-G2B","UBA9145","UBA1172","UBA3478","HIMB100","UBA1268","Casp-actino5","VMDI01","HIMB30","UBA7428","MedAcidi-G2A","Marinobacterium","MED-G52","VMDJ01","WTJO01","UBA11606","Casp-actino5","CG-Epi1","MGIIb-N1","MGIIb-N1","None","Rubripirellula","Marinobacterium","MED-G16","UBA10364","UBA12191","UBA10364","UBA2284","Synechococcus","Poseidonia","UBA724","UBA1014","None","GCA-2728275","MGIIb-N1","None","MGIIa-K1","LFER01","UBA7428","UBA11606","JCVI-SCAAA005","None","None","Marinisoma","UBA8309","Glaciecola","CACIJG01","Poseidonia","Actinomarina","MGIIa-K1","MGIIa-K1","UBA1100","UBA226","MGIIa-K1","UBA8653","MGIIa-L1","UBA3537","EC70","UBA974","GCA-2862225","Litoricola","None","UBA11654","GCA-2718035","MGIIb-O3","MED-G52","MGIIb-P","HIMB30","UBA952","MGIIa-K2","None","Cycloclasticus","UBA11663","Synechococcus","UBA3478","UBA8752","UBA10364","VMDJ01","MED-G13","Actinomarina","GCA-2716345","VMDH01","Luminiphilus","GCA-002701455","UBA7428","LFER01","UBA8752"],"t_species":["WTJO01 sp011524805","UBA685 sp011524995","None","UBA3478 sp011525015","None","RFYP01 sp011526365","BACL21 sp011525065","UBA8309 sp001627655","Coraliomargarita akajimensis","None","UBA7971 sp002686885","MedAcidi-G2B sp003215175","None","UBA1172 sp011522975","UBA3478 sp003045935","HIMB100 sp002722385","UBA1268 sp002694955","None","VMDI01 sp008080915","None","UBA7428 sp011526285","MedAcidi-G2A sp002380145","Marinobacterium sp013283365","MED-G52 sp002457055","VMDJ01 sp008080925","WTJO01 sp011524805","UBA11606 sp002694825","Casp-actino5 sp010024485","CG-Epi1 sp002724775","MGIIb-N1 sp011523055","MGIIb-N1 sp002505455","None","Rubripirellula sp012270385","Marinobacterium sp013373575","MED-G16 sp002701745","UBA10364 sp002378005","UBA12191 sp012269595","None","UBA2284 sp003605085","None","Poseidonia sp011524855","UBA724 sp002723075","UBA1014 sp001469005","None","GCA-2728275 sp002728275","None","None","None","LFER01 sp001510135","UBA7428 sp011526285","UBA11606 sp002694825","JCVI-SCAAA005 sp011522845","None","None","None","UBA8309 sp002686685","Glaciecola sp000155775","CACIJG01 sp002703515","Poseidonia sp002495645","None","None","MGIIa-K1 sp002498185","UBA1100 sp011523245","UBA226 sp002505935","None","UBA8653 sp002705085","MGIIa-L1 sp011526375","UBA3537 sp001735715","EC70 sp002696885","None","GCA-2862225 sp002862225","Litoricola sp012269655","None","UBA11654 sp001629325","GCA-2718035 sp002718035","None","MED-G52 sp001627375","MGIIb-P sp002498455","None","UBA952 sp011522905","MGIIa-K2 sp002719395","None","Cycloclasticus sp002700385","UBA11663 sp011524845","None","UBA3478 sp003045935","UBA8752 sp002862785","None","VMDJ01 sp008080925","MED-G13 sp902602415","None","GCA-2716345 sp902597195","VMDH01 sp008080945","None","GCA-002701455 sp902578675","UBA7428 sp011526275","LFER01 sp011525185","None"]},"columns":[{"accessor":"bin name","name":"MAG ID","type":"character","sortable":true,"resizable":true,"filterable":true,"header":"MAG ID","align":"left","footerStyle":{"fontWeight":"bold"},"minWidth":150,"sticky":"left","style":{"borderRight":"5px solid #eee"},"headerStyle":{"borderRight":"5px solid #eee"}},{"accessor":"domain","name":"domain","type":"character","sortable":true,"resizable":true,"filterable":true,"header":"domain","align":"left","footerStyle":{"fontWeight":"bold"}},{"accessor":"confidence","name":"confidence","type":"numeric","sortable":true,"resizable":true,"filterable":false,"header":"confidence","align":"left","footerStyle":{"fontWeight":"bold"}},{"accessor":"% completion","name":"% completion","type":"numeric","sortable":true,"resizable":true,"filterable":false,"header":"% completion","align":"left","footerStyle":{"fontWeight":"bold"}},{"accessor":"% redundancy","name":"% redundancy","type":"numeric","sortable":true,"resizable":true,"filterable":false,"header":"% redundancy","align":"left","footerStyle":{"fontWeight":"bold"}},{"accessor":"num_splits","name":"num_splits","type":"numeric","sortable":true,"resizable":true,"filterable":false,"header":"num splits","align":"left","footerStyle":{"fontWeight":"bold"}},{"accessor":"total length","name":"total length","type":"numeric","sortable":true,"resizable":true,"filterable":false,"header":"total length","align":"left","footerStyle":{"fontWeight":"bold"},"style":{"borderRight":"5px solid #eee"},"headerStyle":{"borderRight":"5px solid #eee"}},{"accessor":"total_scgs","name":"total_scgs","type":"numeric","sortable":true,"resizable":true,"filterable":true,"header":"total scgs","align":"left","footerStyle":{"fontWeight":"bold"}},{"accessor":"supporting_scgs","name":"supporting_scgs","type":"numeric","sortable":true,"resizable":true,"filterable":true,"header":"supporting scgs","align":"left","footerStyle":{"fontWeight":"bold"}},{"accessor":"t_domain","name":"t_domain","type":"character","sortable":true,"resizable":true,"filterable":true,"header":"t domain","align":"left","footerStyle":{"fontWeight":"bold"}},{"accessor":"t_phylum","name":"t_phylum","type":"character","sortable":true,"resizable":true,"filterable":true,"header":"t phylum","align":"left","footerStyle":{"fontWeight":"bold"}},{"accessor":"t_class","name":"t_class","type":"character","sortable":true,"resizable":true,"filterable":true,"header":"t class","align":"left","footerStyle":{"fontWeight":"bold"}},{"accessor":"t_order","name":"t_order","type":"character","sortable":true,"resizable":true,"filterable":true,"header":"t order","align":"left","footerStyle":{"fontWeight":"bold"}},{"accessor":"t_family","name":"t_family","type":"character","sortable":true,"resizable":true,"filterable":true,"header":"t family","align":"left","footerStyle":{"fontWeight":"bold"}},{"accessor":"t_genus","name":"t_genus","type":"character","sortable":true,"resizable":true,"filterable":true,"header":"t genus","align":"left","footerStyle":{"fontWeight":"bold"}},{"accessor":"t_species","name":"t_species","type":"character","sortable":true,"resizable":true,"filterable":true,"header":"t species","align":"left","footerStyle":{"fontWeight":"bold"}}],"columnGroups":[{"name":"Completeness","headerStyle":{"fontSize":"1em","borderRight":"5px solid #eee"},"columns":["domain","confidence","% completion","% redundancy","num_splits","total length"]},{"name":"SCG Taxonomy","headerStyle":{"fontSize":"1em"},"columns":["total_scgs","supporting_scgs","t_domain","t_phylum","t_class","t_order","t_family","t_genus","t_species"]}],"searchable":true,"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,98],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"062c155f73e65353543ec91babac5a6b"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
<div class="infobox">
<div class="center">
<p><i class="fas fa-file"></i> <strong>Data pack file</strong></p>
</div>
<p>06_MAG_TAXONOMY_and_COMPLETENESS.txt</a></p>
</div>
<p>Ok, let’s review.</p>
<p>In total, our collection has 98 MAGs. According to <code>anvi-estimate-genome-completeness</code> there were 80 Bacteria MAGs, 17 Archaea MAGs, and 1 Eukarya MAG.</p>
<p>The values were a little bit lower according to <code>anvi-estimate-scg-taxonomy</code>, specifically 77 Bacteria MAGs, 16 Archaea MAGs, and no Eukarya MAGs.</p>
<p>So why the difference? Well first of all the GTDB does not contain Eukaryotes. Next, take a look at the <code>total scgs</code> column from the <code>anvi-estimate-scg-taxonomy</code> output (see table above) and notice that 5 MAGs have the value 0. This indicates that of the 22 SCG used for taxonomic assignment, these MAGs have no hits.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>What else? Well, there are 73 MAGs that have completion values &gt; 70% and redundancy values &lt; 10% and 70 that have at least 10 total SCG used in taxonomic classification.</p>
<blockquote>
<p>The last thing to check is whether the domain classification is the same for <code>anvi-estimate-genome-completeness</code> and <code>anvi-estimate-scg-taxonomy</code> (where applicable).</p>
</blockquote>
<p>Well we know there are 98 MAGs total, 5 of which have no GTDB classification. So, there should be 93 MAGs that have identical domain classification. Let’s see (and hope so).</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>tmp_tbl <span class="ot">&lt;-</span> mag_summary</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>tmp_tbl<span class="sc">$</span>t_domain <span class="ot">&lt;-</span> <span class="fu">toupper</span>(tmp_tbl<span class="sc">$</span>t_domain)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(tmp_tbl<span class="sc">$</span>domain <span class="sc">==</span> tmp_tbl<span class="sc">$</span>t_domain)</span></code></pre></div>
<pre><code>[1] 93</code></pre>
</div>
<p>Booya.</p>
<h2 id="scg-hits-across-mags">SCG Hits across MAGs</h2>
<p>Next we can look at the distribution of SCGs from the <code>Bacteria_71</code> and <code>Archaea_76</code> HMM profiles across the MAGs. For this we first need to generate individual <code>PROFILE.db</code>s and <code>CONTIG.db</code>s for each MAG in our curate MAG collection. Luckily, anvi’o has us covered.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-split</span> <span class="at">--pan-or-profile-db</span> 06_MERGED/WA/PROFILE.db <span class="dt">\</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">--contigs-db</span> 03_CONTIGS/WA-contigs.db  <span class="dt">\</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">--collection-name</span> DASTOOL_MAGS  <span class="dt">\</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">--output-dir</span> 10_SPLITS/WA/</span></code></pre></div>
<p>As part of this program, anvi’o will add a collection called <code>DEFAULT</code> and a single bin called <code>ALL_SPLITS</code>. This will become important in the next step when we generate an <code>internal-genomes</code> file.</p>
<p>The program we will use to look at the distribution of SCGs across MAGs is <code>anvi-script-gen-hmm-hits-matrix-across-genomes</code>. The program needs either an <code>external-genomes</code> file or an <code>internal-genomes</code> file so it knows where to find the genomes of interest for the analysis. We will use the <code>internal-genomes</code> file, which is a five-column TAB-delimited flat text file.</p>
<p>The header line must contain these columns: <strong>‘name’, ‘bin_id’, ‘collection_id’, ‘profile_db_path’, ‘contigs_db_path’</strong>. Each line lists a single MAG. As far as I know the <code>external-genomes</code> option also works perfectly fine.</p>
<p>So we generate this file and run the command against both HMM profiles.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-script-gen-hmm-hits-matrix-across-genomes</span> <span class="dt">\</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>                 <span class="at">--internal-genomes</span> internal-genomes.txt <span class="dt">\</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">--hmm-source</span> Archaea_76 <span class="dt">\</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">--output-file</span> 07_Archaea_76-across-MAGs.txt</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-script-gen-hmm-hits-matrix-across-genomes</span> <span class="dt">\</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">--internal-genomes</span> internal-genomes.txt <span class="dt">\</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">--hmm-source</span> Bacteria_71 <span class="dt">\</span></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>                 <span class="at">--output-file</span> 07_Bacteria_71-across-MAGs.txt</span></code></pre></div>
<div class="infobox">
<div class="center">
<p><i class="fas fa-file"></i> <strong>Data pack files</strong></p>
</div>
<p>07_Archaea_76-across-MAGs.txt &amp; 07_Bacteria_71-across-MAGs.txt</a></p>
</div>
<p>Sweet, now for both HMM profiles, we can look at <strong>a</strong>) the number of MAGs represented by each SCG and <strong>b</strong>) the number of unique SCG present in each MAG. Of course keep in mind that we ran both HMM profiles across all MAGs, <em>regardless of their domain classification</em>. So what this means for example is that we expect an Archaea MAG to have fewer hits when run against <code>Bacteria_71</code> than against <code>Archaea_76</code>.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p></br></p>
<h3 id="summary-of-bacteria-71-scg">Summary of Bacteria 71 SCG</h3>
<div class="l-body">
<div class="panelset">
<section id="by-scg" class="panel">
<h4>By SCG</h4>
<p><small><strong>Table 4 | Total number of MAGs containing each gene in the Bacteria_71 HMM profile. </strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-c8a4d3306664d068b980" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-c8a4d3306664d068b980">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"SCG":["ADK","AICARFT_IMPCHas","ATP-synt","ATP-synt_A","Adenylsucc_synt","Chorismate_synt","EF_TS","Exonuc_VII_L","GrpE","Ham1p_like","IPPT","OSCP","PGK","Pept_tRNA_hydro","RBFA","RNA_pol_L","RNA_pol_Rpb6","RRF","RecO_C","Ribonuclease_P","Ribosom_S12_S23","Ribosomal_L1","Ribosomal_L13","Ribosomal_L14","Ribosomal_L16","Ribosomal_L17","Ribosomal_L18p","Ribosomal_L19","Ribosomal_L2","Ribosomal_L20","Ribosomal_L21p","Ribosomal_L22","Ribosomal_L23","Ribosomal_L27","Ribosomal_L27A","Ribosomal_L28","Ribosomal_L29","Ribosomal_L3","Ribosomal_L32p","Ribosomal_L35p","Ribosomal_L4","Ribosomal_L5","Ribosomal_L6","Ribosomal_L9_C","Ribosomal_S10","Ribosomal_S11","Ribosomal_S13","Ribosomal_S15","Ribosomal_S16","Ribosomal_S17","Ribosomal_S19","Ribosomal_S2","Ribosomal_S20p","Ribosomal_S3_C","Ribosomal_S6","Ribosomal_S7","Ribosomal_S8","Ribosomal_S9","RsfS","RuvX","SecE","SecG","SecY","SmpB","TsaE","UPF0054","YajC","eIF-1a","ribosomal_L24","tRNA-synt_1d","tRNA_m1G_MT"],"total_mags":[76,75,69,65,76,79,69,55,74,82,62,62,80,57,60,85,75,70,54,49,67,81,77,81,86,68,73,64,80,60,62,81,82,60,89,61,82,82,49,53,83,87,87,62,74,82,86,75,64,83,82,82,61,83,61,72,85,79,62,61,60,58,87,61,63,63,54,77,72,80,65]},"columns":[{"accessor":"SCG","name":"SCG","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["ADK","AICARFT_IMPCHas","ATP-synt","ATP-synt_A","Adenylsucc_synt","Chorismate_synt","EF_TS","Exonuc_VII_L","GrpE","Ham1p_like","IPPT","OSCP","PGK","Pept_tRNA_hydro","RBFA","RNA_pol_L","RNA_pol_Rpb6","RRF","RecO_C","Ribonuclease_P","Ribosom_S12_S23","Ribosomal_L1","Ribosomal_L13","Ribosomal_L14","Ribosomal_L16","Ribosomal_L17","Ribosomal_L18p","Ribosomal_L19","Ribosomal_L2","Ribosomal_L20","Ribosomal_L21p","Ribosomal_L22","Ribosomal_L23","Ribosomal_L27","Ribosomal_L27A","Ribosomal_L28","Ribosomal_L29","Ribosomal_L3","Ribosomal_L32p","Ribosomal_L35p","Ribosomal_L4","Ribosomal_L5","Ribosomal_L6","Ribosomal_L9_C","Ribosomal_S10","Ribosomal_S11","Ribosomal_S13","Ribosomal_S15","Ribosomal_S16","Ribosomal_S17","Ribosomal_S19","Ribosomal_S2","Ribosomal_S20p","Ribosomal_S3_C","Ribosomal_S6","Ribosomal_S7","Ribosomal_S8","Ribosomal_S9","RsfS","RuvX","SecE","SecG","SecY","SmpB","TsaE","UPF0054","YajC","eIF-1a","ribosomal_L24","tRNA-synt_1d","tRNA_m1G_MT"],"header":"SCG","align":"left","footerStyle":{"fontWeight":"bold"},"minWidth":100,"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"total_mags","name":"total_mags","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["76","75","69","65","76","79","69","55","74","82","62","62","80","57","60","85","75","70","54","49","67","81","77","81","86","68","73","64","80","60","62","81","82","60","89","61","82","82","49","53","83","87","87","62","74","82","86","75","64","83","82","82","61","83","61","72","85","79","62","61","60","58","87","61","63","63","54","77","72","80","65"],"header":"total mags","align":"left","footerStyle":{"fontWeight":"bold"}}],"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,71],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"4883bcb6bca5e79f60c300d47e3f34a5"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</section>
<section id="by-mag" class="panel">
<h4>By MAG</h4>
<p><small><strong>Table 5 | Total number of SCG from the Bacteria_71 HMM profile found in each MAG. </strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-e942d34f3ba649439361" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-e942d34f3ba649439361">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"bin name":["WA_MAG_00001","WA_MAG_00002","WA_MAG_00003","WA_MAG_00004","WA_MAG_00005","WA_MAG_00006","WA_MAG_00007","WA_MAG_00008","WA_MAG_00009","WA_MAG_00010","WA_MAG_00011","WA_MAG_00012","WA_MAG_00013","WA_MAG_00014","WA_MAG_00015","WA_MAG_00016","WA_MAG_00017","WA_MAG_00018","WA_MAG_00019","WA_MAG_00020","WA_MAG_00021","WA_MAG_00022","WA_MAG_00023","WA_MAG_00024","WA_MAG_00025","WA_MAG_00026","WA_MAG_00027","WA_MAG_00028","WA_MAG_00029","WA_MAG_00030","WA_MAG_00031","WA_MAG_00032","WA_MAG_00033","WA_MAG_00034","WA_MAG_00035","WA_MAG_00036","WA_MAG_00037","WA_MAG_00038","WA_MAG_00039","WA_MAG_00040","WA_MAG_00041","WA_MAG_00042","WA_MAG_00043","WA_MAG_00044","WA_MAG_00045","WA_MAG_00046","WA_MAG_00047","WA_MAG_00048","WA_MAG_00049","WA_MAG_00050","WA_MAG_00051","WA_MAG_00052","WA_MAG_00053","WA_MAG_00054","WA_MAG_00055","WA_MAG_00056","WA_MAG_00057","WA_MAG_00058","WA_MAG_00059","WA_MAG_00060","WA_MAG_00061","WA_MAG_00062","WA_MAG_00063","WA_MAG_00064","WA_MAG_00065","WA_MAG_00066","WA_MAG_00067","WA_MAG_00068","WA_MAG_00069","WA_MAG_00070","WA_MAG_00071","WA_MAG_00072","WA_MAG_00073","WA_MAG_00074","WA_MAG_00075","WA_MAG_00076","WA_MAG_00077","WA_MAG_00078","WA_MAG_00079","WA_MAG_00080","WA_MAG_00081","WA_MAG_00082","WA_MAG_00083","WA_MAG_00084","WA_MAG_00085","WA_MAG_00086","WA_MAG_00087","WA_MAG_00088","WA_MAG_00089","WA_Bin_00090","WA_Bin_00091","WA_Bin_00093","WA_Bin_00094","WA_Bin_00095","WA_Bin_00097","WA_Bin_00098","WA_Bin_00099","WA_Bin_00105"],"domain":["BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","EUKARYA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","ARCHAEA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","ARCHAEA","ARCHAEA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","BACTERIA","ARCHAEA","ARCHAEA","BACTERIA","ARCHAEA","ARCHAEA","BACTERIA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","BACTERIA","ARCHAEA","BACTERIA","BACTERIA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA"],"unique_scgs":[71,71,70,69,69,69,70,70,68,62,68,68,67,69,69,69,67,66,69,65,70,66,65,65,67,68,66,64,33,35,36,67,64,63,64,64,63,60,60,63,35,61,62,59,28,33,34,34,57,58,61,60,60,61,59,56,60,60,31,59,35,30,58,33,34,55,26,52,52,50,50,49,47,51,51,22,48,26,46,47,23,39,38,36,36,21,21,22,20,47,45,46,47,41,37,37,25,23]},"columns":[{"accessor":"bin name","name":"MAG","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["WA_MAG_00001","WA_MAG_00002","WA_MAG_00003","WA_MAG_00004","WA_MAG_00005","WA_MAG_00006","WA_MAG_00007","WA_MAG_00008","WA_MAG_00009","WA_MAG_00010","WA_MAG_00011","WA_MAG_00012","WA_MAG_00013","WA_MAG_00014","WA_MAG_00015","WA_MAG_00016","WA_MAG_00017","WA_MAG_00018","WA_MAG_00019","WA_MAG_00020","WA_MAG_00021","WA_MAG_00022","WA_MAG_00023","WA_MAG_00024","WA_MAG_00025","WA_MAG_00026","WA_MAG_00027","WA_MAG_00028","WA_MAG_00029","WA_MAG_00030","WA_MAG_00031","WA_MAG_00032","WA_MAG_00033","WA_MAG_00034","WA_MAG_00035","WA_MAG_00036","WA_MAG_00037","WA_MAG_00038","WA_MAG_00039","WA_MAG_00040","WA_MAG_00041","WA_MAG_00042","WA_MAG_00043","WA_MAG_00044","WA_MAG_00045","WA_MAG_00046","WA_MAG_00047","WA_MAG_00048","WA_MAG_00049","WA_MAG_00050","WA_MAG_00051","WA_MAG_00052","WA_MAG_00053","WA_MAG_00054","WA_MAG_00055","WA_MAG_00056","WA_MAG_00057","WA_MAG_00058","WA_MAG_00059","WA_MAG_00060","WA_MAG_00061","WA_MAG_00062","WA_MAG_00063","WA_MAG_00064","WA_MAG_00065","WA_MAG_00066","WA_MAG_00067","WA_MAG_00068","WA_MAG_00069","WA_MAG_00070","WA_MAG_00071","WA_MAG_00072","WA_MAG_00073","WA_MAG_00074","WA_MAG_00075","WA_MAG_00076","WA_MAG_00077","WA_MAG_00078","WA_MAG_00079","WA_MAG_00080","WA_MAG_00081","WA_MAG_00082","WA_MAG_00083","WA_MAG_00084","WA_MAG_00085","WA_MAG_00086","WA_MAG_00087","WA_MAG_00088","WA_MAG_00089","WA_Bin_00090","WA_Bin_00091","WA_Bin_00093","WA_Bin_00094","WA_Bin_00095","WA_Bin_00097","WA_Bin_00098","WA_Bin_00099","WA_Bin_00105"],"header":"MAG","align":"left","footerStyle":{"fontWeight":"bold"},"minWidth":100,"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"domain","name":"domain","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","EUKARYA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","ARCHAEA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","ARCHAEA","ARCHAEA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","BACTERIA","ARCHAEA","ARCHAEA","BACTERIA","ARCHAEA","ARCHAEA","BACTERIA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","BACTERIA","ARCHAEA","BACTERIA","BACTERIA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA"],"header":"domain","align":"left","footerStyle":{"fontWeight":"bold"}},{"accessor":"unique_scgs","name":"unique_scgs","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["71","71","70","69","69","69","70","70","68","62","68","68","67","69","69","69","67","66","69","65","70","66","65","65","67","68","66","64","33","35","36","67","64","63","64","64","63","60","60","63","35","61","62","59","28","33","34","34","57","58","61","60","60","61","59","56","60","60","31","59","35","30","58","33","34","55","26","52","52","50","50","49","47","51","51","22","48","26","46","47","23","39","38","36","36","21","21","22","20","47","45","46","47","41","37","37","25","23"],"header":"unique scgs","align":"left","footerStyle":{"fontWeight":"bold"}}],"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,98],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"461b070bcdfa3f49b2c204d7bc4c5033"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</section>
</div>
</div>
<h3 id="summary-of-archaea-76-scgs">Summary of Archaea 76 SCGs</h3>
<div class="l-body">
<div class="panelset">
<section id="by-scg-1" class="panel">
<h4>By SCG</h4>
<p><small><strong>Table 6 | Total number of MAGs containing each gene in the Archaea_76 HMM profile. </strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-43a61f41b73160818200" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-43a61f41b73160818200">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"SCG":["ATP-synt_D","ATP-synt_F","Adenylsucc_synt","AdoHcyase","Archease","CTP-dep_RFKase","CarS-like","DNA_primase_lrg","DUF357","DUF359","DUF655","Diphthamide_syn","FbpA","HMG-CoA_red","Ham1p_like","NDK","PPS_PS","PTH2","Prefoldin","PyrI","RNA_pol_L_2","RNA_pol_N","RNA_pol_Rpb4","RNA_pol_Rpb6","RNase_HII","Rib_5-P_isom_A","Ribosom_S12_S23","Ribosomal_L1","Ribosomal_L13","Ribosomal_L14","Ribosomal_L15e","Ribosomal_L16","Ribosomal_L21e","Ribosomal_L22","Ribosomal_L23","Ribosomal_L26","Ribosomal_L29","Ribosomal_L3","Ribosomal_L31e","Ribosomal_L32e","Ribosomal_L37ae","Ribosomal_L39","Ribosomal_L4","Ribosomal_L44","Ribosomal_L5e","Ribosomal_L6","Ribosomal_S11","Ribosomal_S13","Ribosomal_S15","Ribosomal_S17","Ribosomal_S17e","Ribosomal_S19","Ribosomal_S19e","Ribosomal_S2","Ribosomal_S24e","Ribosomal_S27e","Ribosomal_S28e","Ribosomal_S3Ae","Ribosomal_S7","Ribosomal_S8","Ribosomal_S8e","Ribosomal_S9","RtcB","SecY","Spt4","TIM","Trm56","TruD","V_ATPase_I","dsDNA_bind","eIF-6","tRNA-synt_1c","tRNA-synt_1d","tRNA-synt_His","vATP-synt_AC39","vATP-synt_E"],"total_mags":[16,14,76,66,14,15,20,16,14,16,12,16,21,35,82,73,13,17,18,14,15,16,2,75,80,34,67,81,77,81,12,86,12,81,82,13,82,82,14,16,13,3,83,16,15,87,82,86,75,83,15,82,17,82,14,10,15,15,72,85,17,79,23,87,14,69,13,27,17,8,13,88,80,75,14,15]},"columns":[{"accessor":"SCG","name":"SCG","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["ATP-synt_D","ATP-synt_F","Adenylsucc_synt","AdoHcyase","Archease","CTP-dep_RFKase","CarS-like","DNA_primase_lrg","DUF357","DUF359","DUF655","Diphthamide_syn","FbpA","HMG-CoA_red","Ham1p_like","NDK","PPS_PS","PTH2","Prefoldin","PyrI","RNA_pol_L_2","RNA_pol_N","RNA_pol_Rpb4","RNA_pol_Rpb6","RNase_HII","Rib_5-P_isom_A","Ribosom_S12_S23","Ribosomal_L1","Ribosomal_L13","Ribosomal_L14","Ribosomal_L15e","Ribosomal_L16","Ribosomal_L21e","Ribosomal_L22","Ribosomal_L23","Ribosomal_L26","Ribosomal_L29","Ribosomal_L3","Ribosomal_L31e","Ribosomal_L32e","Ribosomal_L37ae","Ribosomal_L39","Ribosomal_L4","Ribosomal_L44","Ribosomal_L5e","Ribosomal_L6","Ribosomal_S11","Ribosomal_S13","Ribosomal_S15","Ribosomal_S17","Ribosomal_S17e","Ribosomal_S19","Ribosomal_S19e","Ribosomal_S2","Ribosomal_S24e","Ribosomal_S27e","Ribosomal_S28e","Ribosomal_S3Ae","Ribosomal_S7","Ribosomal_S8","Ribosomal_S8e","Ribosomal_S9","RtcB","SecY","Spt4","TIM","Trm56","TruD","V_ATPase_I","dsDNA_bind","eIF-6","tRNA-synt_1c","tRNA-synt_1d","tRNA-synt_His","vATP-synt_AC39","vATP-synt_E"],"header":"SCG","align":"left","footerStyle":{"fontWeight":"bold"},"minWidth":100,"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"total_mags","name":"total_mags","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["16","14","76","66","14","15","20","16","14","16","12","16","21","35","82","73","13","17","18","14","15","16","2","75","80","34","67","81","77","81","12","86","12","81","82","13","82","82","14","16","13","3","83","16","15","87","82","86","75","83","15","82","17","82","14","10","15","15","72","85","17","79","23","87","14","69","13","27","17","8","13","88","80","75","14","15"],"header":"total mags","align":"left","footerStyle":{"fontWeight":"bold"}}],"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,76],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"a0c55b228a83da4ee5e1621cbe0d713b"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</section>
<section id="by-mag-1" class="panel">
<h4>By MAG</h4>
<p><small><strong>Table 6 | Total number of MAGs containing each gene in the Archaea_76 HMM profile. </strong></small></p>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-c794c77de7fe106baa2f" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-c794c77de7fe106baa2f">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"bin name":["WA_MAG_00001","WA_MAG_00002","WA_MAG_00003","WA_MAG_00004","WA_MAG_00005","WA_MAG_00006","WA_MAG_00007","WA_MAG_00008","WA_MAG_00009","WA_MAG_00010","WA_MAG_00011","WA_MAG_00012","WA_MAG_00013","WA_MAG_00014","WA_MAG_00015","WA_MAG_00016","WA_MAG_00017","WA_MAG_00018","WA_MAG_00019","WA_MAG_00020","WA_MAG_00021","WA_MAG_00022","WA_MAG_00023","WA_MAG_00024","WA_MAG_00025","WA_MAG_00026","WA_MAG_00027","WA_MAG_00028","WA_MAG_00029","WA_MAG_00030","WA_MAG_00031","WA_MAG_00032","WA_MAG_00033","WA_MAG_00034","WA_MAG_00035","WA_MAG_00036","WA_MAG_00037","WA_MAG_00038","WA_MAG_00039","WA_MAG_00040","WA_MAG_00041","WA_MAG_00042","WA_MAG_00043","WA_MAG_00044","WA_MAG_00045","WA_MAG_00046","WA_MAG_00047","WA_MAG_00048","WA_MAG_00049","WA_MAG_00050","WA_MAG_00051","WA_MAG_00052","WA_MAG_00053","WA_MAG_00054","WA_MAG_00055","WA_MAG_00056","WA_MAG_00057","WA_MAG_00058","WA_MAG_00059","WA_MAG_00060","WA_MAG_00061","WA_MAG_00062","WA_MAG_00063","WA_MAG_00064","WA_MAG_00065","WA_MAG_00066","WA_MAG_00067","WA_MAG_00068","WA_MAG_00069","WA_MAG_00070","WA_MAG_00071","WA_MAG_00072","WA_MAG_00073","WA_MAG_00074","WA_MAG_00075","WA_MAG_00076","WA_MAG_00077","WA_MAG_00078","WA_MAG_00079","WA_MAG_00080","WA_MAG_00081","WA_MAG_00082","WA_MAG_00083","WA_MAG_00084","WA_MAG_00085","WA_MAG_00086","WA_MAG_00087","WA_MAG_00088","WA_MAG_00089","WA_Bin_00090","WA_Bin_00091","WA_Bin_00093","WA_Bin_00094","WA_Bin_00095","WA_Bin_00097","WA_Bin_00098","WA_Bin_00099","WA_Bin_00105"],"domain":["BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","EUKARYA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","ARCHAEA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","ARCHAEA","ARCHAEA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","BACTERIA","ARCHAEA","ARCHAEA","BACTERIA","ARCHAEA","ARCHAEA","BACTERIA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","BACTERIA","ARCHAEA","BACTERIA","BACTERIA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA"],"unique_scgs":[31,31,32,32,31,31,32,30,31,54,29,31,32,31,32,30,33,31,32,31,32,31,30,31,30,29,31,29,67,67,70,32,25,30,30,30,32,31,28,31,69,29,30,28,63,64,62,64,26,26,27,28,28,30,26,24,28,24,61,27,61,60,28,61,55,27,55,26,23,25,26,25,23,22,22,51,20,45,23,21,48,21,19,14,20,10,5,9,4,26,22,22,25,24,20,24,12,14]},"columns":[{"accessor":"bin name","name":"MAG","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["WA_MAG_00001","WA_MAG_00002","WA_MAG_00003","WA_MAG_00004","WA_MAG_00005","WA_MAG_00006","WA_MAG_00007","WA_MAG_00008","WA_MAG_00009","WA_MAG_00010","WA_MAG_00011","WA_MAG_00012","WA_MAG_00013","WA_MAG_00014","WA_MAG_00015","WA_MAG_00016","WA_MAG_00017","WA_MAG_00018","WA_MAG_00019","WA_MAG_00020","WA_MAG_00021","WA_MAG_00022","WA_MAG_00023","WA_MAG_00024","WA_MAG_00025","WA_MAG_00026","WA_MAG_00027","WA_MAG_00028","WA_MAG_00029","WA_MAG_00030","WA_MAG_00031","WA_MAG_00032","WA_MAG_00033","WA_MAG_00034","WA_MAG_00035","WA_MAG_00036","WA_MAG_00037","WA_MAG_00038","WA_MAG_00039","WA_MAG_00040","WA_MAG_00041","WA_MAG_00042","WA_MAG_00043","WA_MAG_00044","WA_MAG_00045","WA_MAG_00046","WA_MAG_00047","WA_MAG_00048","WA_MAG_00049","WA_MAG_00050","WA_MAG_00051","WA_MAG_00052","WA_MAG_00053","WA_MAG_00054","WA_MAG_00055","WA_MAG_00056","WA_MAG_00057","WA_MAG_00058","WA_MAG_00059","WA_MAG_00060","WA_MAG_00061","WA_MAG_00062","WA_MAG_00063","WA_MAG_00064","WA_MAG_00065","WA_MAG_00066","WA_MAG_00067","WA_MAG_00068","WA_MAG_00069","WA_MAG_00070","WA_MAG_00071","WA_MAG_00072","WA_MAG_00073","WA_MAG_00074","WA_MAG_00075","WA_MAG_00076","WA_MAG_00077","WA_MAG_00078","WA_MAG_00079","WA_MAG_00080","WA_MAG_00081","WA_MAG_00082","WA_MAG_00083","WA_MAG_00084","WA_MAG_00085","WA_MAG_00086","WA_MAG_00087","WA_MAG_00088","WA_MAG_00089","WA_Bin_00090","WA_Bin_00091","WA_Bin_00093","WA_Bin_00094","WA_Bin_00095","WA_Bin_00097","WA_Bin_00098","WA_Bin_00099","WA_Bin_00105"],"header":"MAG","align":"left","footerStyle":{"fontWeight":"bold"},"minWidth":100,"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"accessor":"domain","name":"domain","type":"character","sortable":true,"resizable":true,"filterable":true,"cell":["BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","EUKARYA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","ARCHAEA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","ARCHAEA","ARCHAEA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","BACTERIA","ARCHAEA","ARCHAEA","BACTERIA","ARCHAEA","ARCHAEA","BACTERIA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","ARCHAEA","BACTERIA","ARCHAEA","BACTERIA","BACTERIA","ARCHAEA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA","BACTERIA"],"header":"domain","align":"left","footerStyle":{"fontWeight":"bold"}},{"accessor":"unique_scgs","name":"unique_scgs","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["31","31","32","32","31","31","32","30","31","54","29","31","32","31","32","30","33","31","32","31","32","31","30","31","30","29","31","29","67","67","70","32","25","30","30","30","32","31","28","31","69","29","30","28","63","64","62","64","26","26","27","28","28","30","26","24","28","24","61","27","61","60","28","61","55","27","55","26","23","25","26","25","23","22","22","51","20","45","23","21","48","21","19","14","20","10","5","9","4","26","22","22","25","24","20","24","12","14"],"header":"unique scgs","align":"left","footerStyle":{"fontWeight":"bold"}}],"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,98],"paginationType":"numbers","showPageInfo":true,"minRows":1,"highlight":true,"bordered":true,"striped":true,"nowrap":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"550b573c27b013989847d0075f397c60"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</section>
</div>
</div>
<h2 id="scg-coverage-across-mags">SCG Coverage across MAGs</h2>
<p>In the section above we used <code>anvi-split</code> to generate <code>PROFILE.db</code>s and <code>CONTIG.db</code>s for each MAG. We will now query these databases to access gene hits to the Archaea_76 and Bacteria_71 HMM profiles.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> mag <span class="kw">in</span> <span class="kw">`</span><span class="fu">cat</span> 00_WA-MAGS.txt<span class="kw">`</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span> <span class="ex">anvi-script-get-hmm-hits-per-gene-call</span> <span class="dt">\</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">--contigs-db</span> 10_SPLITS/WA/<span class="va">$mag</span>/CONTIGS.db <span class="dt">\</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">--hmm-source</span> Archaea_76 <span class="dt">\</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>                       <span class="at">--output-file</span> Archaea_76/09_HMM_HITS_PER_GENE_CALLS/<span class="va">$mag</span>-Archaea_76.txt</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span> <span class="ex">anvi-script-get-hmm-hits-per-gene-call</span> <span class="dt">\</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>                       <span class="at">--contigs-db</span> 10_SPLITS/WA/<span class="va">$mag</span>/CONTIGS.db <span class="dt">\</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">--hmm-source</span> Bacteria_71 <span class="dt">\</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>                       <span class="at">--output-file</span> Bacteria_71/09_HMM_HITS_PER_GENE_CALLS/<span class="va">$mag</span>-Bacteria_71.txt</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>These commands will each generate a 5-column text file for every MAG containing the following information:</p>
<ul>
<li><p><code>unique_entry_id</code> I honestly don’t know what this is but we do not need this info here (FIX).</p></li>
<li><p><code>gene_callers_id</code> The unique gene caller ID recognized by anvi’o.</p></li>
<li><p><code>source</code> The name of the functional annotation source, in other words the database that the function came from. In this case, either Arachea_76 or Bacteria_71.</p></li>
<li><p><code>gene_name</code> The name of the SCG.</p></li>
<li><p><code>gene_hmm_id</code> The Pfam gene ID.</p></li>
</ul>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>tmp_hits <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">&quot;files/MAG/Bacteria_71/09_HMM_HITS_PER_GENE_CALLS/WA_MAG_00001-Bacteria_71.txt&quot;</span>, <span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(tmp_hits)</span></code></pre></div>
<pre><code># A tibble: 6 × 5
  unique_entry_id gene_callers_id source      gene_name    gene_hmm_id
            &lt;dbl&gt;           &lt;dbl&gt; &lt;chr&gt;       &lt;chr&gt;        &lt;chr&gt;      
1           14261         1055959 Bacteria_71 PGK          PF00162.19 
2           14262         1055973 Bacteria_71 IPPT         PF01715.17 
3           14265         1056198 Bacteria_71 Ribosomal_S… PF00312.22 
4           14266         1056200 Bacteria_71 RBFA         PF02033.18 
5           14267         1056209 Bacteria_71 UPF0054      PF02130.17 
6           14308         1057505 Bacteria_71 Ribosomal_S7 PF00177.21 </code></pre>
</div>
<p>What we need to do is parse out the <code>gene_callers_id</code> column and remove the header. A simple <code>bash</code> one-liner should do the trick. We will use the file <code>00_WA-MAGS.txt</code>–which is a single column file containing the names of each MAG–to loop through each output file and generate gene ID files.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> mag <span class="kw">in</span> <span class="kw">`</span><span class="fu">cat</span> 00_WA-MAGS.txt<span class="kw">`</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span> <span class="fu">awk</span> <span class="st">&#39;{print $2}&#39;</span> files/MAG/Archaea_76/09_HMM_HITS_PER_GENE_CALLS/<span class="va">$mag</span>-Archaea_76.txt <span class="kw">|</span> <span class="fu">sed</span> 1d <span class="op">&gt;</span> Archaea_76/10_GENE_CALLER_IDS/<span class="va">$mag</span>-Archaea_76-gene_ids.txt</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span> <span class="fu">awk</span> <span class="st">&#39;{print $2}&#39;</span> files/MAG/Bacteria_71/09_HMM_HITS_PER_GENE_CALLS/<span class="va">$mag</span>-Bacteria_71.txt <span class="kw">|</span> <span class="fu">sed</span> 1d <span class="op">&gt;</span> Bacteria_71/10_GENE_CALLER_IDS/<span class="va">$mag</span>-Bacteria_71-gene_ids.txt</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>Excellent. Then we use the <code>-gene_ids.txt</code> files in conjunction with <code>anvi-export-gene-coverage-and-detection</code> to generate coverage and detection information for each MAG from both HMM profiles. Here we use the list of MAGs (<code>00_WA-MAGS.txt</code>) in a <code>for</code> loop to produce the files for each MAG.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> mag <span class="kw">in</span> <span class="kw">`</span><span class="fu">cat</span> 00_WA-MAGS.txt<span class="kw">`</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span> <span class="ex">anvi-export-gene-coverage-and-detection</span> <span class="dt">\</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">--profile-db</span> <span class="va">$mag</span>/PROFILE.db <span class="dt">\</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">--contigs-db</span> <span class="va">$mag</span>/CONTIGS.db <span class="dt">\</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">--genes-of-interest</span> <span class="va">$mag</span>-Archaea_76-gene_ids.txt <span class="dt">\</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">--output-file-prefix</span> <span class="va">$mag</span>-Archaea_76</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span> <span class="ex">anvi-export-gene-coverage-and-detection</span> <span class="dt">\</span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">--profile-db</span> <span class="va">$mag</span>/PROFILE.db <span class="dt">\</span></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">--contigs-db</span> <span class="va">$mag</span>/CONTIGS.db <span class="dt">\</span></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">--genes-of-interest</span> <span class="va">$mag</span>-Bacteria_71-gene_ids.txt <span class="dt">\</span></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">--output-file-prefix</span> <span class="va">$mag</span>-Bacteria_71</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>It is worth noting that the same data can be generated by adding the <code>-init-gene-coverages</code> flag to <code>anvi-summarize</code> on the collection of MAGs. This will generate coverage stats for all genes in a MAG. At that point you would need to parse out the genes of interest from the output files. Story for another time.</p>
<p>Almost done. It would be nice to add in a few more details for each SCG hit from our MAGs. We can use <code>anvi-get-sequences-for-hmm-hits</code> for this task. The output of this function is a fasta file containing detailed deflines. For the record, the deflines contains information on the source HMM profile, e-value score, parent contig id, gene callers id, gene name, gene length, and the gene’s start/stop positions. Here we use the <code>seqinr</code> package to read the fasta file and then parse out the relevant information.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-get-sequences-for-hmm-hits</span> <span class="at">--internal-genomes</span> internal-genomes.txt <span class="dt">\</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">--hmm-sources</span> Archaea_76 <span class="dt">\</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">--output-file</span> 08_MAGS-Archaea_76-sequences.fa <span class="dt">\</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>                                <span class="at">--unique-genes</span> <span class="at">--no-wrap</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-get-sequences-for-hmm-hits</span> <span class="at">--internal-genomes</span> internal-genomes.txt </span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>                                <span class="ex">--hmm-sources</span> Bacteria_71 <span class="dt">\</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>                                <span class="at">--output-file</span> 08_MAGS-Bacteria_71-sequences.fa <span class="dt">\</span></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>                                <span class="at">--unique-genes</span> <span class="at">--no-wrap</span></span></code></pre></div>
<div class="infobox">
<div class="center">
<p><i class="fas fa-file"></i> <strong>Data pack files</strong></p>
</div>
<p>08_MAGS-Archaea_76-sequences.fa &amp; 08_MAGS-Bacteria_71-sequences.fa</a></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>tmp_fa <span class="ot">&lt;-</span> seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="st">&quot;files/MAG/08_MAGS-Archaea_76-sequences.fa&quot;</span>, </span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">whole.header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>tmp_fa <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(seqinr<span class="sc">::</span><span class="fu">getName</span>(tmp_fa))</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>tmp_fa <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(tmp_fa, <span class="at">X1 =</span> <span class="dv">1</span>)</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="co"># COount number of delimiters (|) and add 1 for first column</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>tmp_ncols <span class="ot">&lt;-</span> <span class="fu">max</span>(stringr<span class="sc">::</span><span class="fu">str_count</span>(tmp_fa<span class="sc">$</span>X1, <span class="st">&quot;[|]&quot;</span>)) <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>tmp_colmn <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;col&quot;</span>, <span class="dv">1</span><span class="sc">:</span>tmp_ncols, <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>tmp_gene_info <span class="ot">&lt;-</span> tidyr<span class="sc">::</span><span class="fu">separate</span>(<span class="at">data =</span> tmp_fa, <span class="at">col =</span> X1, <span class="at">sep =</span> <span class="st">&quot;[|]&quot;</span>, </span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">into =</span> tmp_colmn, <span class="at">remove =</span> <span class="cn">TRUE</span>)</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>tmp_gene_info[, <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">6</span>,<span class="dv">7</span>)] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>tmp_gene_info<span class="sc">$</span>col_5 <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;^gene_callers_id:hash.*_&quot;</span>, <span class="st">&quot;&quot;</span>, tmp_gene_info<span class="sc">$</span>col_5)</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>tmp_gene_info<span class="sc">$</span>col_3 <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;^e_value:&quot;</span>, <span class="st">&quot;&quot;</span>, tmp_gene_info<span class="sc">$</span>col_3)</span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>tmp_gene_info<span class="sc">$</span>col_1 <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_{3}Archaea_76_{3}.*bin_id:&quot;</span>, <span class="st">&quot;XXXXX&quot;</span>, tmp_gene_info<span class="sc">$</span>col_1)</span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>tmp_gene_info<span class="sc">$</span>col_8 <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;^length:&quot;</span>, <span class="st">&quot;&quot;</span>, tmp_gene_info<span class="sc">$</span>col_8)</span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>tmp_gene_info <span class="ot">&lt;-</span> tmp_gene_info <span class="sc">%&gt;%</span> tidyr<span class="sc">::</span><span class="fu">separate</span>(<span class="at">col =</span> col_1, <span class="at">sep =</span> <span class="st">&quot;XXXXX&quot;</span>, </span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&quot;gene_name&quot;</span>, <span class="st">&quot;mag_id&quot;</span>), <span class="at">remove =</span> <span class="cn">TRUE</span>)</span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>tmp_gene_info <span class="ot">&lt;-</span> tmp_gene_info <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">e_value =</span> <span class="dv">3</span>, <span class="at">gene_callers_id =</span> <span class="dv">4</span>, <span class="at">length =</span> <span class="dv">5</span>)  </span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>arch76_gene_hits <span class="ot">&lt;-</span> tmp_gene_info</span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a>arch76_gene_hits<span class="sc">$</span>gene_callers_id <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(arch76_gene_hits<span class="sc">$</span>gene_callers_id)</span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>tmp_fa <span class="ot">&lt;-</span> seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="st">&quot;files/MAG/08_MAGS-Bacteria_71-sequences.fa&quot;</span>, </span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">whole.header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>tmp_fa <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(seqinr<span class="sc">::</span><span class="fu">getName</span>(tmp_fa))</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>tmp_fa <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(tmp_fa, <span class="at">X1 =</span> <span class="dv">1</span>)</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a><span class="co"># COount number of delimiters (|) and add 1 for first column</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>tmp_ncols <span class="ot">&lt;-</span> <span class="fu">max</span>(stringr<span class="sc">::</span><span class="fu">str_count</span>(tmp_fa<span class="sc">$</span>X1, <span class="st">&quot;[|]&quot;</span>)) <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>tmp_colmn <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;col&quot;</span>, <span class="dv">1</span><span class="sc">:</span>tmp_ncols, <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>tmp_gene_info <span class="ot">&lt;-</span> tidyr<span class="sc">::</span><span class="fu">separate</span>(<span class="at">data =</span> tmp_fa, <span class="at">col =</span> X1, <span class="at">sep =</span> <span class="st">&quot;[|]&quot;</span>, </span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">into =</span> tmp_colmn, <span class="at">remove =</span> <span class="cn">TRUE</span>)</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>tmp_gene_info[, <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">6</span>,<span class="dv">7</span>)] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>tmp_gene_info<span class="sc">$</span>col_5 <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;^gene_callers_id:hash.*_&quot;</span>, <span class="st">&quot;&quot;</span>, tmp_gene_info<span class="sc">$</span>col_5)</span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>tmp_gene_info<span class="sc">$</span>col_3 <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;^e_value:&quot;</span>, <span class="st">&quot;&quot;</span>, tmp_gene_info<span class="sc">$</span>col_3)</span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a>tmp_gene_info<span class="sc">$</span>col_1 <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_{3}Bacteria_71_{3}.*bin_id:&quot;</span>, <span class="st">&quot;XXXXX&quot;</span>, tmp_gene_info<span class="sc">$</span>col_1)</span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>tmp_gene_info<span class="sc">$</span>col_8 <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;^length:&quot;</span>, <span class="st">&quot;&quot;</span>, tmp_gene_info<span class="sc">$</span>col_8)</span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a>tmp_gene_info <span class="ot">&lt;-</span> tmp_gene_info <span class="sc">%&gt;%</span> tidyr<span class="sc">::</span><span class="fu">separate</span>(<span class="at">col =</span> col_1, <span class="at">sep =</span> <span class="st">&quot;XXXXX&quot;</span>, </span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&quot;gene_name&quot;</span>, <span class="st">&quot;mag_id&quot;</span>), <span class="at">remove =</span> <span class="cn">TRUE</span>)</span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-20"><a href="#cb23-20" aria-hidden="true" tabindex="-1"></a>tmp_gene_info <span class="ot">&lt;-</span> tmp_gene_info <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">e_value =</span> <span class="dv">3</span>, <span class="at">gene_callers_id =</span> <span class="dv">4</span>, <span class="at">length =</span> <span class="dv">5</span>)  </span>
<span id="cb23-21"><a href="#cb23-21" aria-hidden="true" tabindex="-1"></a>bact71_gene_hits <span class="ot">&lt;-</span> tmp_gene_info</span>
<span id="cb23-22"><a href="#cb23-22" aria-hidden="true" tabindex="-1"></a>bact71_gene_hits<span class="sc">$</span>gene_callers_id <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(bact71_gene_hits<span class="sc">$</span>gene_callers_id)</span>
<span id="cb23-23"><a href="#cb23-23" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="infobox">
<div class="center">
<p><i class="fas fa-file"></i> <strong>Data pack files</strong></p>
</div>
<p>09_MAGS-Archaea_76-details.txt &amp; 09_MAGS-Bacteria_71-details.txt</a></p>
</div>
<p>The analysis yielded 5351 genes from MAGs that had hits to the Bacteria_71 profile.</p>
<p>Finally, we can generate files for each MAG that contains the <strong>a</strong>) gene callers ID’s (these are unique), <strong>b</strong>) HMM source (<code>Archaea_76</code> or <code>Bacteria_71</code>), <strong>c</strong>) gene name, <strong>d</strong>) Pfam gene ID, and <strong>e</strong>) coverage or detection across samples.</p>
<p>First for <code>Archaea_76</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>wa_mags <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">&quot;files/MAG/00_WA-MAGS.txt&quot;</span>, <span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>wa_mags <span class="ot">&lt;-</span> wa_mags<span class="sc">$</span>X1</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (mag <span class="cf">in</span> wa_mags) {</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Reading&quot;</span>, mag, <span class="st">&quot;gene list&quot;</span>)</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  tmp_gene_id <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">paste</span>(<span class="st">&quot;files/MAG/Archaea_76/09_HMM_HITS_PER_GENE_CALLS/&quot;</span>, </span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>                                  mag, <span class="st">&quot;-Archaea_76.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), </span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>                            <span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  tmp_gene_id[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Subsetting&quot;</span>, mag, <span class="st">&quot;from SCG detailed hits file. &quot;</span>)</span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>  tmp_subset <span class="ot">&lt;-</span> <span class="fu">subset</span>(arch76_gene_hits, mag_id <span class="sc">==</span> mag)</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Reading&quot;</span>, mag, <span class="st">&quot;coverage&quot;</span>)</span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>  tmp_coverage <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">paste</span>(<span class="st">&quot;files/MAG/Archaea_76/11_COVERAGE_RAW/&quot;</span>, </span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>                                   mag, <span class="st">&quot;-Archaea_76-GENE-COVERAGES.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), </span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>                             <span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Reading&quot;</span>, mag, <span class="st">&quot;detection&quot;</span>)</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>  tmp_detection <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">paste</span>(<span class="st">&quot;files/MAG/Archaea_76/11_DETECTION_RAW/&quot;</span>, </span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>                                    mag, <span class="st">&quot;-Archaea_76-GENE-DETECTION.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), </span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>                              <span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;MAG&quot;</span>, mag, <span class="st">&quot;gene file has &quot;</span>, <span class="fu">nrow</span>(tmp_gene_id), <span class="st">&quot;entries&quot;</span>, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;MAG&quot;</span>, mag, <span class="st">&quot;coverage file has &quot;</span>, <span class="fu">nrow</span>(tmp_coverage), <span class="st">&quot;entries&quot;</span>, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;MAG&quot;</span>, mag, <span class="st">&quot;detection file has &quot;</span>, <span class="fu">nrow</span>(tmp_detection), <span class="st">&quot;entries&quot;</span>, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;MAG&quot;</span>, mag, <span class="st">&quot;details file has &quot;</span>, <span class="fu">nrow</span>(tmp_subset), <span class="st">&quot;entries&quot;</span>, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Joining&quot;</span>, mag, <span class="st">&quot;gene ID and coverage files&quot;</span>)</span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>  tmp_coverage_join <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(tmp_gene_id, tmp_coverage, </span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;gene_callers_id&quot;</span> <span class="ot">=</span> <span class="st">&quot;key&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>                       dplyr<span class="sc">::</span><span class="fu">full_join</span>(., tmp_subset, </span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;gene_callers_id&quot;</span> <span class="ot">=</span> <span class="st">&quot;gene_callers_id&quot;</span>, </span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>                                               <span class="st">&quot;gene_name&quot;</span> <span class="ot">=</span> <span class="st">&quot;gene_name&quot;</span>))</span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>  tmp_coverage_join <span class="ot">&lt;-</span> tmp_coverage_join <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">relocate</span>(<span class="fu">c</span>(<span class="st">&quot;e_value&quot;</span>, <span class="st">&quot;length&quot;</span>), <span class="at">.after =</span> <span class="st">&quot;gene_hmm_id&quot;</span>)</span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Joining&quot;</span>, mag, <span class="st">&quot;gene ID and detection files&quot;</span>)</span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>  tmp_detection_join <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(tmp_gene_id, tmp_detection, </span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;gene_callers_id&quot;</span> <span class="ot">=</span> <span class="st">&quot;key&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a>                       dplyr<span class="sc">::</span><span class="fu">full_join</span>(., tmp_subset, </span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;gene_callers_id&quot;</span> <span class="ot">=</span> <span class="st">&quot;gene_callers_id&quot;</span>, </span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a>                                               <span class="st">&quot;gene_name&quot;</span> <span class="ot">=</span> <span class="st">&quot;gene_name&quot;</span>))</span>
<span id="cb24-36"><a href="#cb24-36" aria-hidden="true" tabindex="-1"></a>  tmp_detection_join <span class="ot">&lt;-</span> tmp_detection_join <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">relocate</span>(<span class="fu">c</span>(<span class="st">&quot;e_value&quot;</span>, <span class="st">&quot;length&quot;</span>), <span class="at">.after =</span> <span class="st">&quot;gene_hmm_id&quot;</span>)</span>
<span id="cb24-37"><a href="#cb24-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-38"><a href="#cb24-38" aria-hidden="true" tabindex="-1"></a>  tmp_coverage_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(mag, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_Archaea_76-coverage.txt&quot;</span>))</span>
<span id="cb24-39"><a href="#cb24-39" aria-hidden="true" tabindex="-1"></a>  tmp_detection_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(mag, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_Archaea_76-detection.txt&quot;</span>))  </span>
<span id="cb24-40"><a href="#cb24-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-41"><a href="#cb24-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(tmp_coverage_join, <span class="fu">paste</span>(<span class="st">&quot;files/MAG/Archaea_76/12_COVERAGE/&quot;</span>, tmp_coverage_name, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), </span>
<span id="cb24-42"><a href="#cb24-42" aria-hidden="true" tabindex="-1"></a>              <span class="at">col.names  =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb24-43"><a href="#cb24-43" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-44"><a href="#cb24-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(tmp_detection_join, <span class="fu">paste</span>(<span class="st">&quot;files/MAG/Archaea_76/12_DETECTION/&quot;</span>, tmp_detection_name, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), </span>
<span id="cb24-45"><a href="#cb24-45" aria-hidden="true" tabindex="-1"></a>              <span class="at">col.names  =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb24-46"><a href="#cb24-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb24-47"><a href="#cb24-47" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<p>And then for <code>Bacteria_71</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>wa_mags <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">&quot;files/MAG/00_WA-MAGS.txt&quot;</span>, <span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>wa_mags <span class="ot">&lt;-</span> wa_mags<span class="sc">$</span>X1</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (mag <span class="cf">in</span> wa_mags) {</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Reading&quot;</span>, mag, <span class="st">&quot;gene list&quot;</span>)</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  tmp_gene_id <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">paste</span>(<span class="st">&quot;files/MAG/Bacteria_71/09_HMM_HITS_PER_GENE_CALLS/&quot;</span>, </span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>                                  mag, <span class="st">&quot;-Bacteria_71.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), </span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>                            <span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>  tmp_gene_id[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Subsetting&quot;</span>, mag, <span class="st">&quot;from SCG detailed hits file. &quot;</span>)</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>  tmp_subset <span class="ot">&lt;-</span> <span class="fu">subset</span>(bact71_gene_hits, mag_id <span class="sc">==</span> mag)</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Reading&quot;</span>, mag, <span class="st">&quot;coverage&quot;</span>)</span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>  tmp_coverage <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">paste</span>(<span class="st">&quot;files/MAG/Bacteria_71/11_COVERAGE_RAW/&quot;</span>, </span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>                                   mag, <span class="st">&quot;-Bacteria_71-GENE-COVERAGES.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), </span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>                             <span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Reading&quot;</span>, mag, <span class="st">&quot;detection&quot;</span>)</span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>  tmp_detection <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">paste</span>(<span class="st">&quot;files/MAG/Bacteria_71/11_DETECTION_RAW/&quot;</span>, </span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>                                    mag, <span class="st">&quot;-Bacteria_71-GENE-DETECTION.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), </span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>                              <span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;MAG&quot;</span>, mag, <span class="st">&quot;gene file has &quot;</span>, <span class="fu">nrow</span>(tmp_gene_id), <span class="st">&quot;entries&quot;</span>, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;MAG&quot;</span>, mag, <span class="st">&quot;coverage file has &quot;</span>, <span class="fu">nrow</span>(tmp_coverage), <span class="st">&quot;entries&quot;</span>, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;MAG&quot;</span>, mag, <span class="st">&quot;detection file has &quot;</span>, <span class="fu">nrow</span>(tmp_detection), <span class="st">&quot;entries&quot;</span>, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;MAG&quot;</span>, mag, <span class="st">&quot;details file has &quot;</span>, <span class="fu">nrow</span>(tmp_subset), <span class="st">&quot;entries&quot;</span>, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Joining&quot;</span>, mag, <span class="st">&quot;gene ID and coverage files&quot;</span>)</span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a>  tmp_coverage_join <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(tmp_gene_id, tmp_coverage, </span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;gene_callers_id&quot;</span> <span class="ot">=</span> <span class="st">&quot;key&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a>                       dplyr<span class="sc">::</span><span class="fu">full_join</span>(., tmp_subset, </span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;gene_callers_id&quot;</span> <span class="ot">=</span> <span class="st">&quot;gene_callers_id&quot;</span>, </span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a>                                               <span class="st">&quot;gene_name&quot;</span> <span class="ot">=</span> <span class="st">&quot;gene_name&quot;</span>))</span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a>  tmp_coverage_join <span class="ot">&lt;-</span> tmp_coverage_join <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">relocate</span>(<span class="fu">c</span>(<span class="st">&quot;e_value&quot;</span>, <span class="st">&quot;length&quot;</span>), <span class="at">.after =</span> <span class="st">&quot;gene_hmm_id&quot;</span>)</span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Joining&quot;</span>, mag, <span class="st">&quot;gene ID and detection files&quot;</span>)</span>
<span id="cb25-31"><a href="#cb25-31" aria-hidden="true" tabindex="-1"></a>  tmp_detection_join <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(tmp_gene_id, tmp_detection, </span>
<span id="cb25-32"><a href="#cb25-32" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;gene_callers_id&quot;</span> <span class="ot">=</span> <span class="st">&quot;key&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb25-33"><a href="#cb25-33" aria-hidden="true" tabindex="-1"></a>                       dplyr<span class="sc">::</span><span class="fu">full_join</span>(., tmp_subset, </span>
<span id="cb25-34"><a href="#cb25-34" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;gene_callers_id&quot;</span> <span class="ot">=</span> <span class="st">&quot;gene_callers_id&quot;</span>, </span>
<span id="cb25-35"><a href="#cb25-35" aria-hidden="true" tabindex="-1"></a>                                               <span class="st">&quot;gene_name&quot;</span> <span class="ot">=</span> <span class="st">&quot;gene_name&quot;</span>))</span>
<span id="cb25-36"><a href="#cb25-36" aria-hidden="true" tabindex="-1"></a>  tmp_detection_join <span class="ot">&lt;-</span> tmp_detection_join <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">relocate</span>(<span class="fu">c</span>(<span class="st">&quot;e_value&quot;</span>, <span class="st">&quot;length&quot;</span>), <span class="at">.after =</span> <span class="st">&quot;gene_hmm_id&quot;</span>)</span>
<span id="cb25-37"><a href="#cb25-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-38"><a href="#cb25-38" aria-hidden="true" tabindex="-1"></a>  tmp_coverage_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(mag, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_Bacteria_71-coverage.txt&quot;</span>))</span>
<span id="cb25-39"><a href="#cb25-39" aria-hidden="true" tabindex="-1"></a>  tmp_detection_name <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map_chr</span>(mag, <span class="sc">~</span> <span class="fu">paste0</span>(., <span class="st">&quot;_Bacteria_71-detection.txt&quot;</span>))  </span>
<span id="cb25-40"><a href="#cb25-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb25-41"><a href="#cb25-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(tmp_coverage_join, <span class="fu">paste</span>(<span class="st">&quot;files/MAG/Bacteria_71/12_COVERAGE/&quot;</span>, tmp_coverage_name, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), </span>
<span id="cb25-42"><a href="#cb25-42" aria-hidden="true" tabindex="-1"></a>              <span class="at">col.names  =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb25-43"><a href="#cb25-43" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb25-44"><a href="#cb25-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(tmp_detection_join, <span class="fu">paste</span>(<span class="st">&quot;files/MAG/Bacteria_71/12_DETECTION/&quot;</span>, tmp_detection_name, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), </span>
<span id="cb25-45"><a href="#cb25-45" aria-hidden="true" tabindex="-1"></a>              <span class="at">col.names  =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span>
<span id="cb25-46"><a href="#cb25-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb25-47"><a href="#cb25-47" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="metagenome-workflow">Metagenome Workflow</h1>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="description-of-data-products-1">Description of Data Products</h2>
<p>The link below gives access to all data and data products generated in the workflow. This is a 25.7Mb zip file.</p>
<div class="layout-chunk" data-layout="l-body">
<a href="https://github.com/istmobiome/scg/raw/main/files/MG_WF_DATA.zip">
<button class="btn btn-success"><i class="fa fa-save"></i> Download the full data pack here</button>
</a>
</div>
<p>There are a number of data products generated in this workflow. Below is a description of each product and links to download the data. Some of this information is redundant, others may not be useful, but all are included here for posterity.</p>
<p>First, a description of individual tables.</p>
<div class="l-body-outset">
<table>
<colgroup>
<col style="width: 64%" />
<col style="width: 35%" />
</colgroup>
<thead>
<tr class="header">
<th><strong>file name</strong></th>
<th><strong>description</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><a href="files/MG/00_SCG_TAXONOMY_GENES.txt">00_SCG_TAXONOMY_GENES.txt</a></td>
<td>list of GTDB SCG.</td>
</tr>
<tr class="even">
<td><a href="files/MG/01_WA-Archaea_76-sequences.fa">01_WA-Archaea_76-sequences.fa</a></td>
<td>fasta file of all Archaea_76 SCG hits.</td>
</tr>
<tr class="odd">
<td><a href="files/MG/01_WA-Bacteria_71-sequences.fa">01_WA-Bacteria_71-sequences.fa</a></td>
<td>fasta file of all Bacteria_71 SCG hits.</td>
</tr>
<tr class="even">
<td><a href="files/MG/02_WA-Archaea_76-HITS.txt">02_WA-Archaea_76-HITS.txt</a></td>
<td>table of all Archaea_76 SCG hits.</td>
</tr>
<tr class="odd">
<td><a href="files/MG/02_WA-Bacteria_71-HITS.txt">02_WA-Bacteria_71-HITS.txt</a></td>
<td>table of all Bacteria_71 SCG hits.</td>
</tr>
<tr class="even">
<td><a href="files/MG/03_WA-Archaea_76-gene_ids.txt">03_WA-Archaea_76-gene_ids.txt</a></td>
<td>gene ids of all Archaea_76 SCG hits.</td>
</tr>
<tr class="odd">
<td><a href="files/MG/03_WA-Bacteria_71-gene_ids.txt">03_WA-Bacteria_71-gene_ids.txt</a></td>
<td>gene ids of all Bacteria_71 SCG hits.</td>
</tr>
<tr class="even">
<td><a href="files/MG/04_WA-Archaea_76-RAW-GENE-COVERAGES.txt">04_WA-Archaea_76-RAW-GENE-COVERAGES.txt</a></td>
<td>raw coverage data for Archaea_76 SCG hits.</td>
</tr>
<tr class="odd">
<td><a href="files/MG/04_WA-Archaea_76-RAW-GENE-DETECTION.txt">04_WA-Archaea_76-RAW-GENE-DETECTION.txt</a></td>
<td>raw detection data for Archaea_76 SCG hits.</td>
</tr>
<tr class="even">
<td><a href="files/MG/04_WA-Bacteria_71-RAW-GENE-COVERAGES.txt">04_WA-Bacteria_71-RAW-GENE-COVERAGES.txt</a></td>
<td>raw coverage data for Bacteria_71 SCG hits.</td>
</tr>
<tr class="odd">
<td><a href="files/MG/04_WA-Bacteria_71-RAW-GENE-DETECTION.txt">04_WA-Bacteria_71-RAW-GENE-DETECTION.txt</a></td>
<td>raw detection data for Bacteria_71 SCG hits.</td>
</tr>
<tr class="even">
<td><a href="files/MG/05_WA-Archaea_76-GENE-COVERAGES.txt">05_WA-Archaea_76-GENE-COVERAGES.txt</a></td>
<td><strong>final coverage data for Archaea_76</strong>.</td>
</tr>
<tr class="odd">
<td><a href="files/MG/05_WA-Archaea_76-GENE-DETECTION.txt">05_WA-Archaea_76-GENE-DETECTION.txt</a></td>
<td><strong>final detection data for Archaea_76</strong>.</td>
</tr>
<tr class="even">
<td><a href="files/MG/05_WA-Bacteria_71-GENE-COVERAGES.txt">05_WA-Bacteria_71-GENE-COVERAGES.txt</a></td>
<td><strong>final coverage data for Bacteria_71</strong>.</td>
</tr>
<tr class="odd">
<td><a href="files/MG/05_WA-Bacteria_71-GENE-DETECTION.txt">05_WA-Bacteria_71-GENE-DETECTION.txt</a></td>
<td><strong>final detection data for Bacteria_71</strong>.</td>
</tr>
<tr class="even">
<td>06_SCG_TAXONOMY/</td>
<td>taxonomy files for each GTDB SCG.</td>
</tr>
<tr class="odd">
<td><a href="files/MG/07_SCG_TAXONOMY.txt">07_SCG_TAXONOMY.txt</a></td>
<td><strong>combined GTDB taxonomy for all genes</strong></td>
</tr>
</tbody>
</table>
</div>
<h2 id="generate-scg-lists-1">Generate SCG lists</h2>
<p>On my first pass through this workflow I noticed that a few genes had hits to several HMM profile genes. So I needed to remove duplicate hits. The best way I found was to use <code>anvi-get-sequences-for-hmm-hits</code> with the flag <code>--unique-genes</code>. From the help menu of this function, anvi’o tells us:</p>
<pre><code>--unique-genes    An HMM source may contain multiple models that can hit
                  the same gene in a given bin or genome. Using this
                  flag, you can ask anvi&#39;o to go through all genes,
                  identify those with multiple hits and report only the
                  most significant hit for each unique gene. (default
                  False)</code></pre>
<p>Excellent. This seemed promising. The output of this function is a fasta file containing detailed deflines.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-get-sequences-for-hmm-hits</span> <span class="at">--contigs-db</span> 03_CONTIGS/WA-contigs.db <span class="dt">\</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">--hmm-sources</span> Archaea_76 <span class="dt">\</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">--output-file</span> 01_WA-Archaea_76-sequences.fa  <span class="dt">\</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>                                <span class="at">--unique-genes</span> <span class="at">--no-wrap</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>                                </span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-get-sequences-for-hmm-hits</span> <span class="at">--contigs-db</span> 03_CONTIGS/WA-contigs.db <span class="dt">\</span></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>                                <span class="at">--hmm-sources</span> Bacteria_71 <span class="dt">\</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>                                <span class="at">--output-file</span> 01_WA-Bacteria_71-sequences.fa  <span class="dt">\</span></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>                                <span class="at">--unique-genes</span> <span class="at">--no-wrap</span></span></code></pre></div>
<div class="infobox">
<div class="center">
<p><i class="fas fa-file"></i> <strong>Data pack files</strong></p>
</div>
<p>01_WA-Archaea_76-sequences.fa &amp; 01_WA-Bacteria_71-sequences.fa</a></p>
</div>
<p>For the record, the deflines contain information on the source HMM profile, e-value score, parent contig id, gene callers id, gene name, gene length, and the gene’s start/stop positions. Here we use the <code>seqinr</code> package to read the fasta file and then I parse out the relevant information.</p>
<p>First, I start with the Archaea_76 profile.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>tmp_fa <span class="ot">&lt;-</span> seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="st">&quot;files/MG/01_WA-Archaea_76-sequences.fa&quot;</span>, </span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">whole.header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>tmp_fa <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(seqinr<span class="sc">::</span><span class="fu">getName</span>(tmp_fa))</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>tmp_fa <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(tmp_fa, <span class="at">X1 =</span> <span class="dv">1</span>)</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="co"># COount number of delimiters (|) and add 1 for first column</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>tmp_ncols <span class="ot">&lt;-</span> <span class="fu">max</span>(stringr<span class="sc">::</span><span class="fu">str_count</span>(tmp_fa<span class="sc">$</span>X1, <span class="st">&quot;[|]&quot;</span>)) <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>tmp_colmn <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;col&quot;</span>, <span class="dv">1</span><span class="sc">:</span>tmp_ncols, <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>tmp_gene_info <span class="ot">&lt;-</span> tidyr<span class="sc">::</span><span class="fu">separate</span>(<span class="at">data =</span> tmp_fa, <span class="at">col =</span> X1, <span class="at">sep =</span> <span class="st">&quot;[|]&quot;</span>, </span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">into =</span> tmp_colmn, <span class="at">remove =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>tmp_gene_info[, <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">6</span>,<span class="dv">7</span>)] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>tmp_gene_info<span class="sc">$</span>col_5 <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;^gene_callers_id:&quot;</span>, <span class="st">&quot;&quot;</span>, tmp_gene_info<span class="sc">$</span>col_5)</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>tmp_gene_info<span class="sc">$</span>col_3 <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;^e_value:&quot;</span>, <span class="st">&quot;&quot;</span>, tmp_gene_info<span class="sc">$</span>col_3)</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>tmp_gene_info<span class="sc">$</span>col_1 <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_{3}.*&quot;</span>, <span class="st">&quot;&quot;</span>, tmp_gene_info<span class="sc">$</span>col_1)</span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>tmp_gene_info<span class="sc">$</span>col_8 <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;^length:&quot;</span>, <span class="st">&quot;&quot;</span>, tmp_gene_info<span class="sc">$</span>col_8)</span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>tmp_gene_info <span class="ot">&lt;-</span> tmp_gene_info <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene_name =</span> <span class="dv">1</span>, <span class="at">e_value =</span> <span class="dv">2</span>, <span class="at">gene_callers_id =</span> <span class="dv">3</span>, <span class="at">length =</span> <span class="dv">4</span>)  </span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>arch76_gene_hits <span class="ot">&lt;-</span> tmp_gene_info</span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>arch76_gene_hits<span class="sc">$</span>gene_callers_id <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(arch76_gene_hits<span class="sc">$</span>gene_callers_id)</span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span></code></pre></div>
</details>
</div>
<p>The analysis yielded 14232 genes that had hits to the Archaea_76 profile. Now for the Bacteria_71 profile.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>tmp_fa <span class="ot">&lt;-</span> seqinr<span class="sc">::</span><span class="fu">read.fasta</span>(<span class="st">&quot;files/MG/01_WA-Bacteria_71-sequences.fa&quot;</span>, </span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">as.string =</span> <span class="cn">TRUE</span>, <span class="at">whole.header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>tmp_fa <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(seqinr<span class="sc">::</span><span class="fu">getName</span>(tmp_fa))</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>tmp_fa <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(tmp_fa, <span class="at">X1 =</span> <span class="dv">1</span>)</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="co"># COount number of delimiters (|) and add 1 for first column</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>tmp_ncols <span class="ot">&lt;-</span> <span class="fu">max</span>(stringr<span class="sc">::</span><span class="fu">str_count</span>(tmp_fa<span class="sc">$</span>X1, <span class="st">&quot;[|]&quot;</span>)) <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>tmp_colmn <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;col&quot;</span>, <span class="dv">1</span><span class="sc">:</span>tmp_ncols, <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>tmp_gene_info <span class="ot">&lt;-</span> tidyr<span class="sc">::</span><span class="fu">separate</span>(<span class="at">data =</span> tmp_fa, <span class="at">col =</span> X1, <span class="at">sep =</span> <span class="st">&quot;[|]&quot;</span>, </span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">into =</span> tmp_colmn, <span class="at">remove =</span> <span class="cn">TRUE</span>)</span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>tmp_gene_info[, <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">6</span>,<span class="dv">7</span>)] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>tmp_gene_info<span class="sc">$</span>col_5 <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;^gene_callers_id:&quot;</span>, <span class="st">&quot;&quot;</span>, tmp_gene_info<span class="sc">$</span>col_5)</span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>tmp_gene_info<span class="sc">$</span>col_3 <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;^e_value:&quot;</span>, <span class="st">&quot;&quot;</span>, tmp_gene_info<span class="sc">$</span>col_3)</span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>tmp_gene_info<span class="sc">$</span>col_1 <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_{3}.*&quot;</span>, <span class="st">&quot;&quot;</span>, tmp_gene_info<span class="sc">$</span>col_1)</span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>tmp_gene_info<span class="sc">$</span>col_8 <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;^length:&quot;</span>, <span class="st">&quot;&quot;</span>, tmp_gene_info<span class="sc">$</span>col_8)</span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>tmp_gene_info <span class="ot">&lt;-</span> tmp_gene_info <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene_name =</span> <span class="dv">1</span>, <span class="at">e_value =</span> <span class="dv">2</span>, <span class="at">gene_callers_id =</span> <span class="dv">3</span>, <span class="at">length =</span> <span class="dv">4</span>)  </span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>bact71_gene_hits <span class="ot">&lt;-</span> tmp_gene_info</span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a>bact71_gene_hits<span class="sc">$</span>gene_callers_id <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(bact71_gene_hits<span class="sc">$</span>gene_callers_id)</span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span></code></pre></div>
</details>
</div>
<p>The analysis yielded 22910 genes that had hits to the Bacteria_71 profile. So for each profile I now had a file containing the gene name, e-value, and unique gene caller id. Next, I run <code>anvi-script-get-hmm-hits-per-gene-call</code>. This function yields basically the same data except it has the duplicate hits and reports the Pfam ID for each gene in the profile. I not only want the Pfam ID but I also want to compare the results as a sanity check and access the number of duplicate hits.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-script-get-hmm-hits-per-gene-call</span> <span class="at">--contigs-db</span> 03_CONTIGS/WA-contigs.db <span class="dt">\</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">--hmm-source</span> Archaea_76 <span class="dt">\</span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">--output-file</span> 02_WA-Archaea_76-HITS.txt</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-script-get-hmm-hits-per-gene-call</span> <span class="at">--contigs-db</span> 03_CONTIGS/WA-contigs.db <span class="dt">\</span></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">--hmm-source</span> Bacteria_71 <span class="dt">\</span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">--output-file</span> 02_WA-Bacteria_71-HITS.txt</span></code></pre></div>
<div class="infobox">
<div class="center">
<p><i class="fas fa-file"></i> <strong>Data pack files</strong></p>
</div>
<p>02_WA-Archaea_76-HITS.txt &amp; 02_WA-Bacteria_71-HITS.txt</a></p>
</div>
<p>Now I can join the two tables by <code>gene_callers_id</code> and <code>gene_name</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>tmp_hits <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">&quot;files/MG/02_WA-Archaea_76-HITS.txt&quot;</span>, <span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>Looks like there were only 4 duplicate hits against the <code>Archaea_76</code> profile.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>tmp_tab <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(arch76_gene_hits, tmp_hits, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;gene_callers_id&quot;</span>, <span class="st">&quot;gene_name&quot;</span>))</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>tmp_tab <span class="ot">&lt;-</span> tmp_tab <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">relocate</span>(<span class="st">&quot;e_value&quot;</span>, <span class="at">.after =</span> <span class="st">&quot;gene_hmm_id&quot;</span>) </span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>tmp_tab <span class="ot">&lt;-</span> tmp_tab <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">relocate</span>(<span class="st">&quot;unique_entry_id&quot;</span>, <span class="at">.before =</span> <span class="st">&quot;gene_name&quot;</span>) </span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>tmp_tab <span class="ot">&lt;-</span> tmp_tab <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">relocate</span>(<span class="st">&quot;source&quot;</span>, <span class="at">.after =</span> <span class="st">&quot;gene_callers_id&quot;</span>) </span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>arch76_gene_info <span class="ot">&lt;-</span> tmp_tab</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>tmp_hits <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">&quot;files/MG/02_WA-Bacteria_71-HITS.txt&quot;</span>, <span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
</details>
</div>
<p>And there were 6 duplicate hits against the <code>Bacteria_71</code> profile.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>tmp_tab <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(bact71_gene_hits, tmp_hits, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;gene_callers_id&quot;</span>, <span class="st">&quot;gene_name&quot;</span>))</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>tmp_tab <span class="ot">&lt;-</span> tmp_tab <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">relocate</span>(<span class="st">&quot;e_value&quot;</span>, <span class="at">.after =</span> <span class="st">&quot;gene_hmm_id&quot;</span>) </span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>tmp_tab <span class="ot">&lt;-</span> tmp_tab <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">relocate</span>(<span class="st">&quot;unique_entry_id&quot;</span>, <span class="at">.before =</span> <span class="st">&quot;gene_name&quot;</span>) </span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>tmp_tab <span class="ot">&lt;-</span> tmp_tab <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">relocate</span>(<span class="st">&quot;source&quot;</span>, <span class="at">.after =</span> <span class="st">&quot;gene_callers_id&quot;</span>) </span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>bact71_gene_info <span class="ot">&lt;-</span> tmp_tab</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span></code></pre></div>
</details>
</div>
<h2 id="scg-coverage">SCG Coverage</h2>
<p>And now we can calculate coverage and detection for each SCG in our analysis. To accomplish this task, we first need to parse out the <code>gene_callers_id</code> column, remove the header, and save the file.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="infobox">
<div class="center">
<p><i class="fas fa-file"></i> <strong>Data pack files</strong></p>
</div>
<p>03_WA-Archaea_76-gene_ids.txt &amp; 03_WA-Bacteria_71-gene_ids.txt</a></p>
</div>
<p>Then run <code>anvi-export-gene-coverage-and-detection</code>.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-export-gene-coverage-and-detection</span> <span class="at">--profile-db</span> 06_MERGED/WA/PROFILE.db <span class="dt">\</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">--contigs-db</span> 03_CONTIGS/WA-contigs.db <span class="dt">\</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">--genes-of-interest</span> WA-Archaea_76-gene_ids.txt <span class="dt">\</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">--output-file-prefix</span> 04_WA-Archaea_76-RAW</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-export-gene-coverage-and-detection</span> <span class="at">--profile-db</span> 06_MERGED/WA/PROFILE.db <span class="dt">\</span></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">--contigs-db</span> 03_CONTIGS/WA-contigs.db <span class="dt">\</span></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">--genes-of-interest</span> WA-Bacteria_71-gene_ids.txt <span class="dt">\</span></span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">--output-file-prefix</span> 04_WA-Bacteria_71-RAW</span></code></pre></div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="infobox">
<div class="center">
<p><i class="fas fa-file"></i> <strong>Data pack file</strong></p>
</div>
<p>04_WA-Archaea_76-RAW-GENE-COVERAGES.txt &amp; 04_WA-Archaea_76-RAW-GENE-DETECTION.txt</a></p>
<p>and</p>
<p>04_WA-Bacteria_71-RAW-GENE-COVERAGES.txt &amp; 04_WA-Bacteria_71-RAW-GENE-DETECTION.txt</a></p>
</div>
<p>Once we run <code>anvi-export-gene-coverage-and-detection</code> for both HMM profiles we can combine these data with all of the other SCG stats generated above.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>tmp_tab <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(arch76_gene_info, arch76_cov, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;gene_callers_id&quot;</span> <span class="ot">=</span> <span class="st">&quot;key&quot;</span>))</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>tmp_tab <span class="ot">&lt;-</span> tmp_tab <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">relocate</span>(<span class="st">&quot;gene_name&quot;</span>, <span class="at">.after =</span> <span class="st">&quot;source&quot;</span>)</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>tmp_tab<span class="sc">$</span>unique_entry_id <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>arch76_cov <span class="ot">&lt;-</span> tmp_tab</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>tmp_tab <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(arch76_gene_info, arch76_det, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;gene_callers_id&quot;</span> <span class="ot">=</span> <span class="st">&quot;key&quot;</span>))</span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>tmp_tab <span class="ot">&lt;-</span> tmp_tab <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">relocate</span>(<span class="st">&quot;gene_name&quot;</span>, <span class="at">.after =</span> <span class="st">&quot;source&quot;</span>)</span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>tmp_tab<span class="sc">$</span>unique_entry_id <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>arch76_det <span class="ot">&lt;-</span> tmp_tab</span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>tmp_tab <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(bact71_gene_info, bact71_cov, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;gene_callers_id&quot;</span> <span class="ot">=</span> <span class="st">&quot;key&quot;</span>))</span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a>tmp_tab <span class="ot">&lt;-</span> tmp_tab <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">relocate</span>(<span class="st">&quot;gene_name&quot;</span>, <span class="at">.after =</span> <span class="st">&quot;source&quot;</span>)</span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a>tmp_tab<span class="sc">$</span>unique_entry_id <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a>bact71_cov <span class="ot">&lt;-</span> tmp_tab</span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb36-18"><a href="#cb36-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-19"><a href="#cb36-19" aria-hidden="true" tabindex="-1"></a>tmp_tab <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(bact71_gene_info, bact71_det, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;gene_callers_id&quot;</span> <span class="ot">=</span> <span class="st">&quot;key&quot;</span>))</span>
<span id="cb36-20"><a href="#cb36-20" aria-hidden="true" tabindex="-1"></a>tmp_tab <span class="ot">&lt;-</span> tmp_tab <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">relocate</span>(<span class="st">&quot;gene_name&quot;</span>, <span class="at">.after =</span> <span class="st">&quot;source&quot;</span>)</span>
<span id="cb36-21"><a href="#cb36-21" aria-hidden="true" tabindex="-1"></a>tmp_tab<span class="sc">$</span>unique_entry_id <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb36-22"><a href="#cb36-22" aria-hidden="true" tabindex="-1"></a>bact71_det <span class="ot">&lt;-</span> tmp_tab</span>
<span id="cb36-23"><a href="#cb36-23" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span></code></pre></div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="infobox">
<div class="center">
<p><i class="fas fa-file"></i> <strong>Data pack file</strong></p>
</div>
<p>05_WA-Archaea_76-GENE-COVERAGES.txt &amp; 05_WA-Archaea_76-GENE-DETECTION.txt</a></p>
<p>and</p>
<p>05_WA-Bacteria_71-GENE-COVERAGES.txt &amp; 05_WA-Bacteria_71-GENE-DETECTION.txt</a></p>
</div>
<h2 id="gtdb-scg-taxonomy">GTDB SCG Taxonomy</h2>
<p>Alright. Now we want the taxonomic assignment for each of the 22 SCG used by the GTDB. For this we use <code>anvi-estimate-scg-taxonomy</code> in <code>--metagenome-mode</code>. When running in metagenome mode, anvi’o <em>automatically chooses the most frequent single-copy core gene</em> to estimate the taxonomic composition within a contigs database. However, we want taxonomic assignment for ALL genes. For this, we generate a list of the GTDB SCG and use a for loop that will run through each SCG and calculate taxonomy for all hits in our database. So this command will produce 22 files.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> gene <span class="kw">in</span> <span class="kw">`</span><span class="ex">SCG_TAXONOMY_GENES.txt</span><span class="kw">`</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span> <span class="ex">anvi-estimate-scg-taxonomy</span>  <span class="at">--contigs-db</span> 03_CONTIGS/WA-contigs.db  <span class="dt">\</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>                               <span class="at">--output-file</span> 06_SCG_TAXONOMY/WA-<span class="va">$gene</span>-TAXONOMY.txt <span class="dt">\</span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>                               <span class="at">--metagenome-mode</span> <span class="dt">\</span></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>                               <span class="at">--scg-name-for-metagenome-mode</span> <span class="va">$gene</span></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<div class="infobox">
<div class="center">
<p><i class="fas fa-file"></i> <strong>Data pack directory</strong></p>
</div>
<p>06_SCG_TAXONOMY/</p>
</div>
<p>Great. Here we use a for loop to <strong>a</strong>) read in each taxonomy file, <strong>b</strong>) do a little reformatting, and <strong>c</strong>) combine all data in one file.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>scg_genes <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">&quot;files/MG/00_SCG_TAXONOMY_GENES.txt&quot;</span>, <span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>scg_tax <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gene <span class="cf">in</span> scg_genes<span class="sc">$</span>X1) {</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>  tmp_tax <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="fu">paste</span>(<span class="st">&quot;files/MG/06_SCG_TAXONOMY/&quot;</span>, <span class="st">&quot;WA-&quot;</span>, gene, <span class="st">&quot;-TAXONOMY.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), </span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>  tmp_tax<span class="sc">$</span>scg_name <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;^WA_&quot;</span>, <span class="st">&quot;&quot;</span>, tmp_tax<span class="sc">$</span>scg_name)</span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>  tmp_tax<span class="sc">$</span>gene_id <span class="ot">&lt;-</span> tmp_tax<span class="sc">$</span>scg_name</span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>  tmp_tax <span class="ot">&lt;-</span> tmp_tax <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">relocate</span>(<span class="st">&quot;gene_id&quot;</span>, <span class="at">.after =</span> <span class="st">&quot;scg_name&quot;</span>)</span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>  tmp_tax<span class="sc">$</span>scg_name <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_[0-9]*$&quot;</span>, <span class="st">&quot;&quot;</span>, tmp_tax<span class="sc">$</span>scg_name)</span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>  tmp_tax<span class="sc">$</span>gene_id <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="fu">paste</span>(gene, <span class="st">&quot;_&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>), <span class="st">&quot;&quot;</span>, tmp_tax<span class="sc">$</span>gene_id)</span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a>  scg_tax <span class="ot">&lt;-</span> <span class="fu">rbind</span>(scg_tax, tmp_tax)</span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">&quot;tmp_&quot;</span>))</span>
<span id="cb38-15"><a href="#cb38-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb38-16"><a href="#cb38-16" aria-hidden="true" tabindex="-1"></a><span class="fu">subset</span>(scg_tax, <span class="fu">duplicated</span>(gene_id))</span>
<span id="cb38-17"><a href="#cb38-17" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(scg_tax, <span class="st">&quot;files/MG/07_SCG_TAXONOMY.txt&quot;</span>, </span>
<span id="cb38-18"><a href="#cb38-18" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</details>
</div>
<div class="infobox">
<div class="center">
<p><i class="fas fa-file"></i> <strong>Data pack file</strong></p>
</div>
<p>07_SCG_TAXONOMY.txt</a></p>
</div>
<details>
<summary>
<strong>Click here</strong> for a breakdown of total hits by GTDB SCG..
</summary>
<div class="layout-chunk" data-layout="l-body">

</div>
<p></br></p>
<table>
<thead>
<tr class="header">
<th><strong>scg_name</strong></th>
<th><strong>count</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Ribosomal_L1</td>
<td>270</td>
</tr>
<tr class="even">
<td>Ribosomal_L13</td>
<td>236</td>
</tr>
<tr class="odd">
<td>Ribosomal_L16</td>
<td>330</td>
</tr>
<tr class="even">
<td>Ribosomal_L17</td>
<td>279</td>
</tr>
<tr class="odd">
<td>Ribosomal_L2</td>
<td>251</td>
</tr>
<tr class="even">
<td>Ribosomal_L20</td>
<td>248</td>
</tr>
<tr class="odd">
<td>Ribosomal_L21p</td>
<td>222</td>
</tr>
<tr class="even">
<td>Ribosomal_L22</td>
<td>269</td>
</tr>
<tr class="odd">
<td>Ribosomal_L27A</td>
<td>187</td>
</tr>
<tr class="even">
<td>Ribosomal_L3</td>
<td>239</td>
</tr>
<tr class="odd">
<td>Ribosomal_L4</td>
<td>197</td>
</tr>
<tr class="even">
<td>Ribosomal_L6</td>
<td>258</td>
</tr>
<tr class="odd">
<td>Ribosomal_L9_C</td>
<td>192</td>
</tr>
<tr class="even">
<td>Ribosomal_S11</td>
<td>270</td>
</tr>
<tr class="odd">
<td>Ribosomal_S2</td>
<td>343</td>
</tr>
<tr class="even">
<td>Ribosomal_S20p</td>
<td>187</td>
</tr>
<tr class="odd">
<td>Ribosomal_S3_C</td>
<td>304</td>
</tr>
<tr class="even">
<td>Ribosomal_S6</td>
<td>195</td>
</tr>
<tr class="odd">
<td>Ribosomal_S7</td>
<td>358</td>
</tr>
<tr class="even">
<td>Ribosomal_S8</td>
<td>278</td>
</tr>
<tr class="odd">
<td>Ribosomal_S9</td>
<td>266</td>
</tr>
<tr class="even">
<td>ribosomal_L24</td>
<td>195</td>
</tr>
</tbody>
</table>
</details>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>That’s all for now. See you soon.</p>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="updates-and-corrections">Corrections</h3>
<p>If you see mistakes or want to suggest changes, please <a href="https://github.com/istmobiome/scg/issues/new">create an issue</a> on the source repository.</p>
<h3 id="reuse">Reuse</h3>
<p>Text and figures are licensed under Creative Commons Attribution <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>. Source code is available at <a href="https://github.com/istmobiome/scg/">https://github.com/istmobiome/scg/</a>, unless otherwise noted. The figures that have been reused from other sources don't fall under this license and can be recognized by a note in their caption: "Figure from ...".</p>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
